<!DOCTYPE html><html><head><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"><script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <script src="scripts/mvc/base-mvc.js" type="text/javascript"></script>
    <script src="scripts/mvc/annotation.js" type="text/javascript"></script>
    <script src="scripts/mvc/tag.js" type="text/javascript"></script>
    <script src="scripts/mvc/tours.js" type="text/javascript"></script>
    <link rel="stylesheet" href="welcome_project.css"><meta charset='UTF-8'><title>coloc-stats: a unified web interface to perform co-localization analysis of genomic features</title></head><body><div class='content'>
    <div class ="tabs">
        <ul class ="nav nav-tabs" id="tab-links">
            <li class="active" value="tab0">
                <a data-toggle="tab" href="#menu0">coloc-stats</a>
            </li>
            <li  value="tab1">
                <a data-toggle="tab" href="#menu1">Documentation</a>
            </li>
            <li  value="tab2">
                <a data-toggle="tab" href="#menu2">FAQ</a>
            </li>
            <li  value="tab3">
                <a data-toggle="tab" href="#menu3">About</a>
            </li>

        </ul>
    </div>
    <div class="tab-content">
        <div id="menu0" class="tab-pane fade  in active">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <h1 class="projectTitle">coloc-stats</h1>
                    <div class = 'projectDescSingle'>
                        a unified web interface to perform <a href="#co-loc-analysis"><font color="#0645AD">co-localization analysis</font></a> of genomic features
                    </div>
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                            <h4>Get started by:</h4>
                            <ol>
                                <li>
                                    <a  class = "button" href="javascript:void(0)" onclick="event.preventDefault(); jQuery('#tool-panel-upload-button', window.parent.jQuery(window.parent.document)).trigger('click');">Uploading datasets</a> of genomic locations (bed-files)
                                </li>
                                <li>
                                    Go to the <a class="button" id="coloc-tool" href="../tool_runner/data_source_redirect?tool_id=hb_conglo_proto_tool">coloc-stats tool</a> to analyze your datasets by multiple methods
                                </li>
                                <li>
                                    View your results (by clicking the eye icon on the history element to the right).
                                </li>
                                <li>
                                    <a href="#" class="tourItem" data-tour.id="coloc.upload">Coloc-stats upload datasets</a>
                                </li>
                            </ol>

                        </div>
                        <div class ="projectDesc" style="margin: 20px 0px; font-style: italic;">
                            If you are new to Galaxy, you can follow a <a class="button" href="https://galaxyproject.org/tutorials/g101/" target="_top">brief tutorial</a>
                        </div>
                        <div class ="projectDesc">
                            <b>You can also:</b>
                            <ul>
                                <li>
                                    <a class="button" href="https://hyperbrowser.uio.no/coloc-stats/u/gsandve/h/coloc-stats-example-analysis" target="_top">View an example analysis</a>
                                </li>
                                <li>
                                    <a class="button" href="https://hyperbrowser.uio.no/coloc-stats/u/gsandve/h/coloc-stats-sample-data" target='_top'>Load a history</a> with small sample datasets (in the next page, click "import history")
                                </li>
                                <li>
                                    Look at the <a href='https://vimeo.com/249248452' target='_top'><font color="#0645AD">screencasts</font></a> below on how to use the system
                                </li>
                            </ul>
                        </div>
                        <div class ="projectDesc" style="margin: 20px 0px; font-style: italic;">
                            Please feel free to <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">contact us</font></a> with any questions.
                        </div>
                        <!--<div class ="projectDesc">-->
                            <!--<p class="paragraph">-->
                                <!--Several generic and specialized tools are available to perform co-localization analysis, they vary in the statistical methodology and analysis approaches, thus potentially providing different conclusions for the same research question. As the findings of co-localization analysis are often the basis for further follow-up experimental investigations, assessing the robustness of the findings is warranted. However, there exists no unified interface to perform co-localization analysis across a multitude of analytical methods and method-specific options (e.g., test statistics, resolution, null models, and whether or not they can handle local genomic heterogeneities and co-localizing confounding features). To address this gap, we propose a coloc-stats, single interface for multiple previously published co-localization analysis methods. </p><p class="paragraph">coloc-stats allows a user to find a method that supports their experimental requirements, and when multiple methods meet the requirements, assess the robustness of conclusions by running and inspecting results across methods. Additionally, we extend methods that only support a pair of tracks to consider track collections.-->
                        <!--</p>-->
                        <!--</div>-->
                    </div>
                </div>
            </div>
        </div>
        <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'example'><h4 class = 'exampleTitle'>Screencast:quick tutorial</h4>
                    <!–– <video controls="controls" width="800" height="600"
                    name="Video Name" src="images/gettingStartedWithColocStats.mov">
                    <!–– </video>
                    <iframe src="https://player.vimeo.com/video/249248452" width="800" height="600" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
                    </div>
                </div>
                </div>
            </div>
        </div>
                        
        <!--<div class="footer">-->
            <!--<div class = 'contactInfo'>-->
                <!--<div class = 'contactInfoDesc'>-->
                    <!--All questions, requests or comments can be redirected to <a href='mailto:geirska@ifi.uio.no' target='_top'>Geir Kjetil Sandve.</a>-->
                <!--</div>-->
            <!--</div>-->
        <!--</div>-->
        </div>



        <div id="menu1" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDescSingle'>
                            <h3>Documentation</h3>
                        </div>
                        <div class = 'projectDescSingle' >
                            <h3># What is co-localization analysis?</h3>
                            <p></p>
                            <div class = 'projectDesc'>
                                <h4 id="co-loc-analysis">Co-localization analysis</h4>
                                <p>High-throughput sequencing methods assay various features of the genome
                                        including regulatory elements, transcription factor binding sites (TFBS),
                                        transcripts and so on. Functionally related genomic features often
                                        co-occur within a genomic sequence (for e.g., co-occurrence of TFBS).
                                        One important way of determining whether genomic features are functionally
                                        related, is to search for their significant co-localization
                                        (based on overlap or proximity). The methodology that determines a
                                        significant co-localization of genomic features is often referred to as
                                        co-localization analysis or alternatively as co-occurrence analysis or
                                    region set enrichment analysis.</p>
                                <p></p>
                                <h4>Basic overview of a typical co-localization analysis</h4>
                                <p>Genomic features are mapped on the reference genome using chromosomal
                                        coordinates (e.g., chr1:100-1000), which are simply the start position
                                        and end position of a sequence of nucleotides on a chromosome. Such a
                                        representation of nucleotide sequence is referred to as genomic regions
                                        or genomic intervals. One of the main outputs of genomic assays is a
                                        list of genomic regions and a collection of sets of genomic regions are
                                        referred to as genomic tracks or region sets. Arithmetic set operations
                                        are performed between genomic tracks to determine the amount of overlap
                                        (overlapping sequence nucleotides) or spatial proximity (e.g. geometric
                                        distance between two genomic regions). This is followed by statistical
                                        testing to determine whether the observed overlap or spatial proximity
                                    is likely due to chance.</p>
                            </div>
                        </div>
                        <!--<div class = 'projectDescSingle'>-->
                            <!--<h3>Input</h3>-->

                            <!--<div class = 'projectDesc'>-->
                                <!--........-->
                            <!--</div>-->
                        <!--</div>-->
                        <!--<div class = 'projectDescSingle'>-->
                            <!--<h3>File formats</h3>-->

                            <!--<div class = 'projectDesc'>-->
                                <!--Coloc-stats supports files only in BED format. BED is a standard file format to store genomic regions, where the first three fields are required to be chr, start and end. Please refer to the BED file format described on the UCSC browser page (https://genome.ucsc.edu/FAQ/FAQformat.html#format1)-->
                            <!--</div>-->
                        <!--</div>-->
                        <!--<div class = 'projectDescSingle'>-->
                            <!--<h3>Invalid file formats</h3>-->

                            <!--<div class = 'projectDesc'>-->
                                <!--Coloc-stats does not accept any other genomic track file formats other than BED format for now. Also, the BED format is required to adhere strictly to the file format specifications.-->
                            <!--</div>-->
                        <!--</div>-->

                        <!--<div class = 'projectDescSingle'>-->
                            <!--<h3>File size limitations</h3>-->
                            <!--<div class = 'projectDesc'>-->
                                <!--We are not aware of any file size limitations as of now.-->
                            <!--</div>-->
                        <!--</div>-->

                        <!--<div class = 'projectDescSingle'>-->
                            <!--<h3>Genome assemblies</h3>-->

                            <!--<div class = 'projectDesc'>-->
                                <!--Coloc-stats does not accept any other genomic track file formats other than BED format for now. Also, the BED format is required to adhere strictly to the file format specifications.-->
                            <!--</div>-->
                        <!--</div>-->


                    </div>
                </div>
            </div>
        </div>
            <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDescSingle'>
                            <h3># What is coloc-stats and what can it do?</h3>
                            <div class = 'projectDesc'>
                                The existing co-localization analysis tools vary in the statistical methodology and analysis approaches,
                                thus potentially providing different conclusions for the same research question. As the findings of
                                co-localization analysis are often the basis for further follow-up experimental investigations,
                                assessing the robustness of the findings is warranted. However, there exists no unified interface
                                to perform co-localization analysis across a multitude of analytical methods and method-specific
                                options (e.g., test statistics, resolution, null models, and whether or not they can handle local
                                genomic heterogeneities and co-localizing confounding features). To address this gap, we propose
                                coloc-stats, a single interface for multiple previously published co-localization analysis methods.
                                <p></p>
                                <div class = 'projectDesc'>
                                        Coloc-stats allows a user:
                                    </div>
                                    <ul>
                                        <li>
                                            To make the most appropriate analysis choices given their research question.
                                        </li>
                                        <li>
                                            Find a relevant method (amongst several) that supports their analysis choices.
                                        </li>
                                        <li>
                                            To assess the robustness of conclusions by running and inspecting results across several methods.
                                        </li>
                                        <li>
                                             Extend methods that only support a pair of tracks to consider track collections.
                                        </li>
                                    </ul>
                            </div>
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>
            <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDescSingle'>
                            <h3># Input, File formats & Genome assemblies</h3>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4> Required files </h4>

                            <div class = 'projectDesc'>
                                For a typical pairwise co-localization analysis, a query track and a reference track are required.
                                The genomic regions obtained through an experiment of your interest is usually used as a query track.
                                Reference track usually contains genomic regions that you hypothesise as having an association with your
                                query track. For example, a query track file could be the genomic regions of SNPs that reached certain
                                significance level in GWAS and a reference track could be the genomic regions of gene promoters.
                                <p></p>
                                It is also possible to run co-localization analysis with one query track against a collection of reference tracks.
                                For example, a query track file could be the genomic regions of SNPs that reached certain significance level in
                                GWAS and the collection of reference tracks could contain the genomic regions of regulatory elements in multiple
                                tissues (multiple tracks). For the ease of users, we provide collections of reference tracks that are of wide interest.
                                We refer to these collections of reference tracks as core databases. Instead of selecting user-specific reference track
                                collections, the user can choose one of the core databases for analysis.
                                <p></p>
                                Some tools (but not all) accept a third file, which typically contains an explicit set of background regions.
                                Such background regions represent the universe of regions that could have possibly ended in the genomic tracks of
                                interest being queried for co-localization. As an example, when testing the co-localization of a SNP set with other
                                annotations, the background set could be all the SNPs covered by the technology platform, which are all assumed to
                                have equal probability to be included in the SNP set of interest. As only a few tools accept this file, this file
                                is not always required.
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4>File formats</h4>

                            <div class = 'projectDesc'>
                                Coloc-stats supports files only in BED format. BED is a standard file format to store genomic regions, where the first
                                three fields are required to be chr, start and end. Please refer to the
                                <a href='https://genome.ucsc.edu/FAQ/FAQformat.html#format1' target='_top'><font color="#0645AD">BED file format</font></a> described on the UCSC browser page.
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4>Invalid file formats</h4>
                            <div class = 'projectDesc'>
                                Coloc-stats does not accept any other genomic track file formats other than BED format for now.
                                Also, the BED format is required to adhere strictly to the file format specifications.
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4>File size limitations</h4>
                            <div class = 'projectDesc'>
                                We did not encounter any file size limitations as of now during several tests.
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4>Genome assemblies</h4>
                            <div class = 'projectDesc'>
                                 Currently hg19, hg38, mm9 and mm10 are supported right away in the coloc-stats tool.
                                However, all the genome assemblies of any species can be used to perform the analysis.
                                To use coloc-stats for a genome assembly or species that is not installed on coloc-stats tool,
                                all you need is a text file containing the chromosome lengths of that genome assembly.
                                Select custom reference genome in the “reference genome” section of the tool, which will
                                prompt you to upload a chromosome lengths file. Upload that file and continue with the remaining
                                analysis choices normally.
                            </div>
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>
            <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDescSingle'>
                            <h3># News</h3>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h4>New features going to be implemented (during Feb-Mar 2018)</h4>
                            <div class = 'projectDesc'>
                                Some of the functionalities mentioned in the below list are implemented only in one or
                                few tools (but not all) integrated in coloc-stats. Thus, we are working on the best ways
                                to provide such functionalities in a unified interface to the users. Nevertheless, many
                                of the features mentioned below are going to be implemented during Feb-Mar 2018.
                                <ul>
                                        <li>
                                            Handling confounding feature tracks
                                        </li>
                                        <li>
                                            Handling local heterogeneities using a third genomic track
                                        </li>
                                        <li>
                                            Heatmaps for many-against-many analyses
                                        </li>
                                        <li>
                                             Counting overlap with customizable flank size
                                        </li>
                                        <li>
                                             Allowing the user to choose the coordinate to be used (start, midpoint or end) when computing distances
                                        </li>
                                        <li>
                                             Allowing the user to choose among different distance metrics (currently only a default distance metric is being used)
                                        </li>
                                        <li>
                                             Allowing the user to select among different types of correlation metrics
                                        </li>
                                    </ul>
                            </div>
                        </div>
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>
        <div id="menu2" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll" style="padding-left: 0px">
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                             <dl id="faqs">
                                 <h3>Frequently asked questions</h3>
                                 (click on the question to see the answer)
                                 <p></p>
                                 <h4 style="margin-left: 40px;">## Why coloc-stats? When coloc-stats? How coloc-stats?</h4>
                                 <p></p>
                                 <ol>
                                     <li>
                                <dt><b>I know some co-localization analysis tools. What is the purpose of a new tool?</b></dt>
                                <dd>
                                    The existing co-localization analysis tools vary in the statistical methodology and analysis approaches,
                                        thus potentially providing different conclusions for the same research question. As the findings of
                                        co-localization analysis are often the basis for further follow-up experimental investigations, assessing
                                        the robustness of the findings is warranted. However, there exists no unified interface to perform co-localization
                                        analysis across a multitude of analytical methods and method-specific options (e.g., test statistics, resolution,
                                        null models, and whether or not they can handle local genomic heterogeneities and co-localizing confounding features).
                                        To address this gap, we propose coloc-stats, a single interface for multiple previously published co-localization analysis methods.
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>What can I achieve more with coloc-stats, than just running other co-localization analysis tools?</b></dt>
                                <dd>
                                    <div class = 'projectDesc'>
                                        Coloc-stats allows a user:
                                    </div>
                                    <ul>
                                        <li>
                                            To make the most appropriate analysis choices given their research question.
                                        </li>
                                        <li>
                                            Find a relevant method (amongst several) that supports their analysis choices.
                                        </li>
                                        <li>
                                            To assess the robustness of conclusions by running and inspecting results across several methods.
                                        </li>
                                        <li>
                                             Extend methods that only support a pair of tracks to consider track collections.
                                        </li>
                                    </ul>
                                </dd>
                                    </li>
                                     <p></p>
                                     <li>
                                <dt><b>When and why should I use coloc-stats?</b></dt>
                                <dd>
                                    <div class = 'projectDesc'>
                                        Coloc-stats is useful when:
                                    </div>
                                    <ul>
                                        <li>
                                            The user is not aware of all the analysis choices that would affect the conclusions of co-localization analysis
                                        </li>
                                        <li>
                                            The user is not familiar of all the existing methods and tools that support their analysis choices and assumptions
                                        </li>
                                        <li>
                                            The user is uncertain about the association between a pair of tracks, to assess the robustness of findings
                                        </li>
                                        <li>
                                             The user wants to run existing pairwise co-localization analysis tools in one-against-many mode
                                        </li>
                                        <li>
                                             The user wants to run several co-localization analysis tools at once
                                        </li>
                                    </ul>
                                </dd>
                                     <p></p>
                                     <li>
                                <dt><b>I do not know how to get started, please help me</b></dt>
                                <dd>
                                To perform co-localization analysis, you need a query track and reference track of interest in
                                    BED format. Once you have the BED files, upload them using the
                                    <a  class = "button" href="javascript:void(0)" onclick="event.preventDefault(); jQuery('#tool-panel-upload-button', window.parent.jQuery(window.parent.document)).trigger('click');">Upload button</a>
                                    on the top left corner of the web page. If you  do not know how to upload files into a galaxy interface,
                                    follow a <a href='https://galaxyproject.org/tutorials/upload/' target='_top'><font color="#0645AD">quick tutorial</font></a>
                                    Once you upload the files, they will appear in your history (on the panel that appears as margin to the right). To know more
                                    about histories in galaxy, <a href='https://galaxyproject.org/tutorials/histories/' target='_top'><font color="#0645AD">see a tutorial here</font></a>.
                                    Then go to the <a class="button" href="../tool_runner/data_source_redirect?tool_id=hb_conglo_proto_tool">coloc-stats tool</a> and choose the relevant
                                    parameter settings to execute. Once you execute the tool, the analysis will be performed by one or several tools in parallel depending upon the chosen
                                    parameter settings. This may take from few minutes to several hours. The results element will be created in your history, which you can also monitor while
                                    the analysis is in progress by clicking on the “eye” icon on the history element. Once all the computations are completed, the history element will turn green.
                                    If the analysis is failed for some reason, it will turn red.

                                </dd>
                                     </li>
                                <p></p>

                                         <p></p>
                                 <h4>## User interface and sharing results</h4>
                                     <p></p>
                                     <li>
                                <dt><b>What is galaxy project ?</b></dt>
                                <dd>
                                If you are not familiar of galaxy project, you can <a href='https://academic.oup.com/nar/article/44/W1/W3/2499339' target='_top'><font color="#0645AD">read a < 10-sentence abstract </font></a>
                                    and visit the <a href='https://usegalaxy.org/' target='_top'><font color="#0645AD">galaxy project homepage</font></a> to know more about it.
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>I am new to a galaxy interface; give me a quick overview on how to navigate.</b></dt>
                                <dd>
                                You can follow a quick and brief tutorial on <a href='https://galaxyproject.org/tutorials/g101/' target='_top'><font color="#0645AD">using a galaxy interface</font></a> (takes ~ 2 min).
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>Is coloc-stats available as a command line tool?</b></dt>
                                <dd>
                                    As of now, coloc-stats is available only through the web server. Command line tool might be a future possibility.
                                </dd>
                                     </li>
                                     <p></p>
                                  <li>
                                <dt><b>Why is it taking ages to run a coloc-stats analysis?</b></dt>
                                <dd>
                                    Depending upon the number of genomic tracks being analysed and also depending upon the number of tools and tool-specific
                                    configuration variants chosen, the analysis might take few minutes to several hours. Especially, some tools by nature are relatively
                                    on a slower side. To quickly confirm that this is not  a server issue, you can run another analysis with less number of genomic tracks
                                    and tools to check if the slowness persists. Depending upon the traffic to the server, sometimes the webpage might appear to be slow and
                                    the computations might be queued. But this is supposed to be a temporary thing and the webpage should return to its original smoothness
                                    within few minutes-hours. If that is not the case <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">you can report to us</font></a> to get support.
                                </dd>
                                     </li>
                                     <p></p>
                                 <li>
                                <dt><b>Is there a way of programmatically accessing coloc-stats page?</b></dt>
                                <dd>
                                    We do not yet have a programming interface, but this will be a possibility in the near future.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>The coloc-stats webpage is quite slow. Is there something I can do to fix it?</b></dt>
                                <dd>
                                    Depending upon the traffic to the server, sometimes the webpage might appear to be slow. But this is supposed to be a temporary thing and the webpage should return to its original smoothness within few minutes-hours.
                                    If that is not the case <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">you can report to us</font></a> to get support.
                                </dd>
                                     </li>
                                     <p></p>
                                     <h4>## Error messages </h4>
                                     <p></p>
                                <li>
                                <dt><b>Something is causing an error, I do not know what it is. What do I do now?</b></dt>
                                <dd>
                                   We observed that some co-localization analysis tools throw errors sometimes because of various different reasons. However, the coloc-stats tool itself should run all the other methods (that pass) and show the results.
                                    If this is not the case, (i.e., if coloc-stats tool itself throws error) then the history element will turn red. When the coloc-stats tool itself throws error, you can  <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">share the history with us</font></a> to get support.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>My input file is causing an error!</b></dt>
                                <dd>
                                    Be aware that coloc-stats currently accepts only BED files. If you have your genomic regions in other file formats, you have to convert them to BED format. BED is a standard file format to store genomic regions, where the first three fields are required to be chr, start and end. Please refer to the
                                    <a href='https://galaxyproject.org/tutorials/g101/' target='_top'><font color="#0645AD">BED file format specifications</font></a> described on the UCSC browser page.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>I am using a BED file as input, still it shows an error!</b></dt>
                                <dd>
                                    Adhering to the BED format specifications is always recommended. Deviations from the file format specifications are a common cause for errors when running several of the bioinformatic tools.
                                    See more about the BED format specifications in the previous question. If you are sure that you adhered to the file format specifications, but still encounter an error, please
                                    <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">report to us</font></a> and we would be very happy to help.
                                </dd>
                                     </li>
                                     <p></p>
                                <h4> ## Results and results pages </h4>
                                     <p></p>
                                 <li>
                                <dt><b>I am not sure if I completely understood the output page. Explain me.</b></dt>
                                <dd>
                                    While the computations of various tools are ongoing, the results page shows statements indicating “in progress”.
                                    Once the computations are completed, the results page is populated with the results from one or several tools,
                                    depending upon the parameters and configurations chosen by the user. The results page shows the parameters chosen
                                    for the analysis. When you run a pairwise co-localization analysis (between two genomic tracks), a plot is displayed
                                    showing the negative logarithmic p-values obtained through different methods/configurations chosen. A simplistic indication
                                    is also provided to the user on how to interpret the findings to allow the user to make an informed decision. A table showing
                                    p-values, enrichment statistics and full results (as given out by the specific tool) of each tool/configuration is also provided.
                                    If any chosen tool fails or reports an error, the error messages are shown in a separate table.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>The links to the full results pages of individual tools are broken. What do I do now?</b></dt>
                                <dd>
                                    This is not the expected behaviour of coloc-stats tool.
                                    Please take a minute to <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">report this to us</font></a>
                                    and we would be very happy to help.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>The p-values are set to NA for some methods. What does this mean?</b></dt>
                                <dd>
                                    Some tools by their nature (e.g., IntervalStats) do not return p-value for the overall association
                                    between tracks as of now. Otherwise all the other tools should return p-value unless they throw an error
                                    for the input genomic tracks. If this is the case, refer to the error messages table to see if an error message
                                    was thrown for the tracks of interest. If you observe a trend of consistent error messages, you can suspect
                                    something went wrong. If that is the case, please take a minute to <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">report this to us</font></a>
                                    and we would be very happy to help.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>Some tables are empty/contain empty fields. What does this mean?</b></dt>
                                <dd>
                                    If no errors are thrown, the table containing error messages would be empty.
                                    This is expected behaviour. Other than that, the tables should not be empty or
                                    contain empty fields. If that is the case, If that is the case, please take a
                                    minute to <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">report this to us</font></a>
                                    and we would be very happy to help.
                                </dd>
                                     </li>
                                     <p></p>
                                <h4> ## Reproducibility, replication and sharing results </h4>
                                     <p></p>
                                 <li>
                                <dt><b>How can I share my analysis history with anyone?</b></dt>
                                <dd>
                                    See here on <a href='https://galaxyproject.org/tutorials/g101/' target='_top'><font color="#0645AD">how to share your analysis history with anyone</font></a>.
                                </dd>
                                     </li>
                                     <p></p>
                                 <li>
                                <dt><b>I was not able to reproduce the analysis that I did earlier.</b></dt>
                                <dd>
                                    All the analyses you performed are stored in the history, if you choose to create a user account and store your analysis histories.
                                    To know more about histories in galaxy, <a href='https://galaxyproject.org/tutorials/histories/' target='_top'><font color="#0645AD">see a brief tutorial </font></a> here.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>I was not able to replicate the findings. The p-values seem to change everytime I run the analysis.</b></dt>
                                <dd>
                                    This is quite expected behaviour in some types of statistical testing that involves permutations and random sampling.
                                    As all the tools integrated into coloc-stats do not support setting a seed, we currently do not provide a functionality to set seed.
                                    On the other hand, if the p-values do not stand significant at a given threshold, that finding should always be interpreted with caution.
                                </dd>
                                     </li>
                                     <p></p>
                                     <h4> ## Genome assemblies and species </h4>
                                     <p></p>
                                <li>
                                <dt><b>Which genome assemblies are supported by coloc-stats?</b></dt>
                                <dd>
                                     Currently hg19, hg38, mm9 and mm10 are supported right away in the coloc-stats tool.
                                    However, all the genome assemblies of any species can be used to perform the analysis.
                                    For that, you only need a text file containing the chromosome lengths.
                                </dd>
                                     </li>
                                     <p></p>
                                <li>
                                <dt><b>How can I use coloc-stats for a different species or genome assembly?</b></dt>
                                <dd>
                                     To use coloc-stats for a genome assembly or species that is not installed on coloc-stats tool,
                                    all you need is a text file containing the chromosome lengths of that genome assembly.
                                    Select custom reference genome in the “reference genome” section of the tool, which will prompt
                                    you to upload a chromosome lengths file. Upload that file and continue with the remaining analysis choices normally.
                                </dd>
                                     </li>
                                     <p></p>
                                     </li>
                            </dl>
<script type="text/javascript">
    $("#faqs dd").hide();
    $("#faqs dt").click(function () {
        $(this).next("#faqs dd").slideToggle(500);
        $(this).toggleClass("expanded");
    });
</script>

                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <div id="menu3" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                            <h4>What is coloc-stats</h4>
                        </div>
                        <div class = 'projectDesc'>
                            Coloc-stats is a collaborative project between the authors of several published co-localization analysis tools. The tool is developed at the Department of Informatics at the University of Oslo and the web server is hosted at https://hyperbrowser.uio.no/coloc-stats/
                        </div>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <h4>Tools integrated in coloc-stats</h4>
                        </div>

                        <div class = 'projectDesc'>
                                Coloc-stats provides a unified interface to eight different co-localization analysis tools. Tools are listed in chronological order they are published.

                        </div>
                        <p></p>
                        <div class = 'description'>
                            <a  target="_blank"  href = 'https://doi.org/10.1186/gb-2010-11-12-r121' >Genomic Hyperbrowser</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank" href = 'https://doi.org/10.1093/bioinformatics/bts009' >IntervalStats</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1371/journal.pcbi.1002529' >GenometriCorr</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1016/j.ajhg.2015.05.016' >GoShifter</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1093/bioinformatics/btv612' >LOLA</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = ' https://doi.org/10.1093/gigascience/gix032' >GSuite Hyperbrowser</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1093/bioinformatics/btx379' >Stereogene</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1101/157735' >GIGGLE</a>
                        </div>


                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<h4>About us</h4>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'projectDesc'>-->
                            <!--<b>Team at the Department of Informatics, University of Oslo</b>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'description'>-->
                            <!--Geir Kjetil Sandve (Principal Investigator)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Boris Simovski-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Chakravarthi Kanduri-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Sveinung Gundersen-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Dmytro Titov-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Diana Domanska-->
                        <!--</div>-->

                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<b>International collaborators</b>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'description'>-->
                            <!--Christoph Bock (CeMM Research Center for Molecular Medicine of the Austrian Academy of Sciences, Vienna, Austria)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Maria Chikina (University of Pittsburgh School of Medicine, Pittsburgh, PA, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Alexander Favorov (John Hopkins University School of Medicine, Baltimore, Maryland, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Ryan Layer (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Aaron Quinlan (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Nathan Sheffield (Center for Public Health Genomics, University of Virginia, Charlottesville, Virginia, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Gosia Trynka (Wellcome Trust Sanger Institute, Hinxton, UK)-->
                        <!--</div>-->

                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<b>Contact us</b>-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--..........-->
                        <!--</div>-->







                    </div>
                </div>
            </div>
        </div>
         <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <h4>About us</h4>
                        </div>
                        <p></p>
                        <div class = 'projectDesc'>
                            <b>Team at the Department of Informatics, University of Oslo</b>
                        </div>
                        <p></p>
                        <div class = 'description'>
                            Geir Kjetil Sandve (Principal Investigator)
                        </div>
                        <div class = 'description'>
                            Boris Simovski
                        </div>
                        <div class = 'description'>
                            Chakravarthi Kanduri
                        </div>
                        <div class = 'description'>
                            Sveinung Gundersen
                        </div>
                        <div class = 'description'>
                            Dmytro Titov
                        </div>
                        <div class = 'description'>
                            Diana Domanska
                        </div>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>International collaborators</b>
                        </div>
                        <p></p>
                        <div class = 'description'>
                            Christoph Bock (CeMM Research Center for Molecular Medicine of the Austrian Academy of Sciences, Vienna, Austria)
                        </div>
                        <div class = 'description'>
                            Maria Chikina (University of Pittsburgh School of Medicine, Pittsburgh, PA, USA)
                        </div>
                        <div class = 'description'>
                            Alexander Favorov (John Hopkins University School of Medicine, Baltimore, Maryland, USA)
                        </div>
                        <div class = 'description'>
                            Ryan Layer (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)
                        </div>
                        <div class = 'description'>
                            Aaron Quinlan (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)
                        </div>
                        <div class = 'description'>
                            Nathan Sheffield (Center for Public Health Genomics, University of Virginia, Charlottesville, Virginia, USA)
                        </div>
                        <div class = 'description'>
                            Gosia Trynka (Wellcome Trust Sanger Institute, Hinxton, UK)
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>
                <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>Contact us</b>
                        </div>
                        <div class = 'description'>
                            For all questions, comments, functionality requests, please <a href='mailto:hyperbrowser-requests@usit.uio.no' target='_top'><font color="#0645AD">contact us</font></a>
                        </div>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>How to cite coloc-stats</b>
                        </div>
                        <div class = 'description'>
                            This information will be updated soon when a preprint is available.
                        </div>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>Version history</b>
                        </div>
                        <div class = 'description'>
                            Current version: 1.0b1
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>



    </div>
</div>
</body>
</html>