<?xml version="1.0" encoding="utf-8" ?>

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.3.9: http://docutils.sourceforge.net/" />
<meta http-equiv="expires" content="Wed, 19 Feb 2003 08:00:00 GMT"/>
<title></title>
<link rel="stylesheet" href="style/welcome.css" type="text/css" />
        <script type="text/javascript" src="scripts/jquery.js"></script>
	    <script src="scripts/jquery_wz_jsgraphics.js" type="text/javascript"></script>
    	<script type="text/javascript" src="scripts/hyperbrowser/flowchart.js"></script>
		<script>
		function getUrlParameter(sParam)
		{
		    var sPageURL = window.location.search.substring(1);
		    var sURLVariables = sPageURL.split('&');
		    for (var i = 0; i < sURLVariables.length; i++)
		    for (var i = 0; i < sURLVariables.length; i++)
		    {
		        var sParameterName = sURLVariables[i].split('=');
		        if (sParameterName[0] == sParam)
		        {
		            return sParameterName[1];
		        }
		    }
		}
		</script>
		
		<script type="text/javascript">
		var jqnc = jQuery.noConflict();
        jqnc(document).ready(function() {
            
		jqnc(".tab4collectionDesc").hide();
		jqnc('.tab4collectionTitle').on('click',function(e) {

		var divid = this.id;
		divid = divid.replace('T','D');
	    jqnc('#'+divid).slideToggle("fast");
	    
	    var divid = this.id;
	    divid = divid.replace('T','B');
	    
	    //console.log(divid, ' ==== ', jqnc('#'+divid).attr("class"));
	    
	    if(jqnc('#'+divid).attr("class") == '-on')
	    {
	    	jqnc('#'+divid).attr('src',"hyperbrowser/images/div/DNAlying.png");
	    	jqnc('#'+divid).attr('class', '-off');
	    	
	    	//console.log(divid, ' 111 ', jqnc('#'+divid).attr("class"));
	    }
	    else
	    {
	    	jqnc('#'+divid).attr('src',"hyperbrowser/images/div/DNAstanding.png");
	    	jqnc('#'+divid).attr('class', '-on');
	    	
	    	//console.log(divid, ' 222 ', jqnc('#'+divid).attr("class"));
	    }
	    
	    
	    
		});
		});
		
		jqnc(document).ready(function() {
            
		jqnc(".sliding").hide();
		jqnc('.show_hide').on('click',function(e) {

		var divid = this.id;
		divid = divid.replace('T','D');
	      
	      jqnc('#'+divid).slideToggle();

			var divid = this.id;
	    divid = divid.replace('T','B');

	    //console.log(divid, ' ==== ', jqnc('#'+divid).attr("class"));

	    if(jqnc('#'+divid).attr("class") == '-on')
	    {
	    	jqnc('#'+divid).attr('src',"hyperbrowser/images/div/DNAlying.png");
	    	jqnc('#'+divid).attr('class', '-off');

	    	//console.log(divid, ' 111 ', jqnc('#'+divid).attr("class"));
	    }
	    else
	    {
	    	jqnc('#'+divid).attr('src',"hyperbrowser/images/div/DNAstanding.png");
	    	jqnc('#'+divid).attr('class', '-on');

	    	//console.log(divid, ' 222 ', jqnc('#'+divid).attr("class"));
	    }


		});
		});
		
		
		
		</script>

        <script type="text/javascript">
            var q = jQuery.noConflict();
            q(document).ready(function() {
              q( "a[minsizehint]" ).click( function() {
                  if ( parent.handle_minwidth_hint ) {
                      parent.handle_minwidth_hint( q(this).attr( "minsizehint" ) );
                  }
              });
			    q('.tabs .tab-links a').on('click', function(e)  {
			        var currentAttrValue = q(this).attr('href');

			        // Show/Hide Tabs
			        q('.tabs ' + currentAttrValue).show().siblings().hide();

			        // Change/remove current tab to active
			        q(this).parent('li').addClass('active').siblings().removeClass('active');

			        e.preventDefault();
			    });

			    /*
			    Open the tab defined by name in the url with parameter tabOn.
			    */
			    var tabOn = getUrlParameter('tabOn');
			    if (tabOn) {
			    	if (tabOn !== 'tab1') {
			    		showDiv('tab1', tabOn)
			    	}
			    }

			    /*
			    Open the basic mode tab (tab2) and the question div defined in the basicQuestionId parameter
			    */
			    var basicQuestionId = getUrlParameter('basicQuestionId');
			    if (basicQuestionId) {
			    	showDiv('tab1', 'tab2')
			    	var qDivId = basicQuestionId.split('_')[0]
			    	toggleDiv('#q' + qDivId, '#i' + qDivId)
			    	var aDivId = basicQuestionId.split('_')[1]
			    	toggleDiv('#a' + qDivId + aDivId, '#i' + qDivId + aDivId)

			    }
            });
        </script>
		<script>
		function showDiv(tabNoVisible, tabVisible) {
		    div1 = document.getElementById(tabVisible);
		    div1.style.display = "block";
		    div2 = document.getElementById(tabNoVisible);
		    div2.style.display = "none";

		    jQuery('#tab-links li[value='+ tabVisible + ']').addClass('active');
		    jQuery('#tab-links li[value='+tabNoVisible+']').removeClass('active');
		}
		</script>
		<script type="text/javascript">
		   if (typeof String.prototype.endsWith !== 'function') {
				String.prototype.endsWith = function(suffix) {
				    return this.indexOf(suffix, this.length - suffix.length) !== -1;
				};
			}
		</script>

		   <script type="text/javascript">
		   var q = jQuery.noConflict();
		   function toggleDiv(divId, imgId) {
         var imgIdFirst = imgId.substr(2,1);
         var imgIdSecond = imgId.substr(3,imgId.length);

			   q(divId).toggle();
			   if (q(imgId).attr("data-open") == 'false') {
				   q(imgId).attr("src", "hyperbrowser/images/div/DNAstanding.png")
				   q(imgId).attr("data-open", "true")

           if(imgId.length === 3)
           {
             window.history.pushState('GSuite', '', 'welcome.html?basicQuestionId=' + imgIdFirst);
           }
           else
           {
             window.history.pushState('GSuite', '', 'welcome.html?basicQuestionId=' + imgIdFirst + '_' + imgIdSecond);
           }
			   }
			   else {
				   q(imgId).attr("src", "hyperbrowser/images/div/DNAlying.png")
				   q(imgId).attr("data-open", "false")

           if(imgId.length === 3)
           {
             window.history.pushState('GSuite', '', 'welcome.html?basicQuestionId=0');
           }
           else
           {
             window.history.pushState('GSuite', '', 'welcome.html?basicQuestionId=' + imgIdFirst);
           }
			   }
		   }

	   </script>

     <script>
      function toggleInfoBox(divId)
      {
        q(divId).toggle();
      }
     </script>

	   <script>
	   function generateAnswerADivContent(divId, toolUrl, toolTitle, trackCollectionExampleText, trackCollectionDataTitle, basicQuestionId ) {
		   var q = jQuery.noConflict();
		   var content = '            <p>\n' +
			   '                This particular question can be seen as a concrete instance of the more general question \n' +
			   '                "How does a single genomic track overlap with various other tracks from a collection", where:	\n' +
			   '            </p>\n' +
			   '                        <ul>\n' +
			   '                <li>\n' +
			   '                    the <span title="a set of genomic regions, e.g. a single bed file"><b>single genomic track</b></span> is the dataset you want to study, which should be uploaded to Galaxy history using e.g. the tool <a href="../tool_runner?tool_id=upload1">Upload File</a> <br><i>(If you want to test with a sample track file, <a href=../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL01TX3JlZ2lvbnNfZXhwYW5kZWRfMTBrYi5iZWQ=&datatype=bed&job_name=sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions&basicQuestionId=' + basicQuestionId + '>click here</a> to load a sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions)</i>\n' +
			   '                </li><br>\n' +
			   '                <li>\n' +
			   '                    a <span title="each track is a set of genomic regions, so that a collection of genomic tracks means multiple datasets, e.g. multiple bed files"><b>track collection</b></span> to be compared against, ' + trackCollectionExampleText + ', which should be constructed using \n' +
			   '                    the tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=' + basicQuestionId + '">Construct track collection (GSuite)</a>\n' +
			   '                </li>\n' +
			   '            </ul>\n' +
			   '            <p>\n' +
			   '                After the necessary data has been collected to Galaxy history, the genomic track of interest can be analyzed against ' + trackCollectionDataTitle + '\n' +
			   '                using the tool <a href="' + toolUrl + '&isBasic=1&isBasic=True"\n' +
			   '                >' + toolTitle +'</a>\n' +
			   '            </p>'

		   q(divId).html(content)

	   }
	   function generateToolHyperlink(anchorId, toolId, title, params, isBasic) {
			var q = jQuery.noConflict();
			hrefVal = '../hyper?mako=generictool&tool_id=' + toolId
			if (isBasic) {
				hrefVal += '&isBasic=1&isBasic=True'
			}
			for (var paramName in params) {
				hrefVal += '&' + paramName + '=' + params[paramName]
			}
//	   		q(anchorId).attr('href', '../hyper?mako=generictool&tool_id=' + toolId + '&basicQuestionId=' + basicQuestionId)
	   		q(anchorId).attr('href', hrefVal)
	   		q(anchorId).html(title)
	   }
	   </script>
	   <script>
	   function draw()
		{
		var canvas = document.getElementById('myCanvas');
		if (canvas.getContext)
		{
			ctx = canvas.getContext('2d');
	
			ctx.strokeStyle = "#696969";
			ctx.strokeRect(480,10,280,730);
			
			text1 = ["Create a GSuite of genomic tracks", "From an External Source",
					 "Create a remote GSuite from a public repository"];
			links1 = ["","","../tool_runner?tool_id=generictool&tool_id=hb_track_source_test_tool"];
			subs1 = [1];
			llength1 = [15,19,21];
			
			createBox(ctx,10,20,60,text1,links1,subs1,llength1,"#EAECEE");
	
			text2 = ["Create a GSuite of genomic tracks", "From History",
					 "Create a GSuite from datasets in your history",
					 "Create a GSuite from a single track based on a single column/attribute",
					 "Create a GSuite from an archive in history"];
			links2 = ["","",
					  "../tool_runner?tool_id=hb_create_g_suite_file_from_history_elements_tool",
					  "../tool_runner?tool_id=hb_compile_gsuite_from_single_track_attribute_tool",
					  "../tool_runner?tool_id=hb_compile_g_suite_from_archive_tool"];
			subs2 = [1];
			llength2 = [15,19,21];
			
			createBox(ctx,10,225,130,text2,links2,subs2,llength2,"#EAECEE");
	
			text3 = ["Create a GSuite of genomic tracks", "From an External Source",
					 "Create a GSuite from an integrated catalog of genomic datasets", "From Hyperbrowser Repository",
					 "Create a GSuite from the HyperBrowser repository"];
			links3 = ["","","../tool_runner?tool_id=hb_track_global_search_tool","",
					  "../tool_runner?tool_id=hb_extract_subtracks_tool"];
			subs3 = [1,3];
			llength3 = [15,19,22];
			
			createBox(ctx,10,525,95,text3,links3,subs3,llength3,"#EAECEE");
	
			text4 = ["Customize a GSuite",
					 "Modify GSuites", "Select a subset of tracks from a GSuite",
					 "Select a subset of meta- data columns in a GSuite",
					 "Concatenate two GSuites", "Edit a metadata column in a GSuite"];
			links4 = ["","","../tool_runner?tool_id=generictool&tool_id=hb_g_suite_select_rows",
					  "../tool_runner?tool_id=hb_g_suite_select_columns",
					  "../tool_runner?tool_id=hb_concatenate_gsuites_tool",
					  "../tool_runner?tool_id=hb_edit_gsuite_metadata_tool"];
			subs4 = [1,6,8];
			llength4 = [15,19,21];
			
			createBox(ctx,170,60,105,text4,links4,subs4,llength4,"#FDF2E9");
	
			text5 = ["Customize a GSuite",
					 "Modify Datasets Referred to in a GSuite", 
					 "Modify primary tracks referred to in a GSuite",
					 "Export GSuite Datasets", 
					 "Export primary tracks from a GSuite to your history"];
			links5 = ["","","../tool_runner?tool_id=hb_g_suite_manipulate_text_files",
					  "","../tool_runner?tool_id=hb_g_suite_export_to_history_tool"];
			subs5 = [1,3];
			llength5 = [15,19,21];
			
			createBox(ctx,170,470,95,text5,links5,subs5,llength5,"#FDF2E9");
	
			text6 = ["Customize a GSuite",
					 "Convert GSuite Tracks", "Convert GSuite tracks from remote to primary"];
			links6 = ["","","../tool_runner?tool_id=hb_g_suite_download_files"];
			subs6 = [1];
			llength6 = [15,19,21];
			
			createBox(ctx,330,20,45,text6,links6,subs6,llength6,"#FDF2E9");
			
			text7 = ["Customize a GSuite",
					 "Convert GSuite Tracks", "Preprocess a GSuite for analysis"];
			links7 = ["","","../tool_runner?tool_id=hb_preprocess_g_suite_tracks_tool"];
			subs7 = [1];
			llength7 = [15,19,21];
			
			createBox(ctx,330,310,45,text7,links7,subs7,llength7,"#FDF2E9");
			
	
	
			text8 = ["Customize a GSuite",
					 "Convert GSuite Tracks",
					 "Convert GSuite a tracks from preprocessed to primary"];
			links8 = ["","",
					  "../tool_runner?tool_id=hb_g_suite_convert_from_preprocessed_to_primary_tool"];
			subs8 = [1];
			llength8 = [15,19,21];
			
			createBox(ctx,330,470,60,text8,links8,subs8,llength8,"#FDF2E9");
	
			text9 = ["Quantitative overview of GSuites",
					 "Metadata Overview", "Summary of metadata attributes for GSuite",
					 "Track Overview", "Summary statistics per track in a GSuite",
					 "Similarity and uniqueness of tracks in GSuite",
					 "Tendency of track elements to clump together along the genome (Ripleys K)",
					 "Calculate all summary statistics for tracks in GSuite (all of the above)",
					 "Compute a basic measure for each track in a GSuite",
					 "Determine suite tracks coinciding with another suite"];
			links9 = ["","","../tool_runner?tool_id=hb_g_suite_metadata_overview_tool",
					  "","../tool_runner?tool_id=hb_pilot_page_basic_overview_of_tracks_in_g_suite_tool",
					  "../tool_runner?tool_id=hb_pilot_page_similarity_and_uniqueness_of_tracks_tool",
					  "../tool_runner?tool_id=hb_pilot_page_clustering_in_tracks_tool",
					  "../tool_runner?tool_id=hb_pilot_page_combined_overview_of_tracks_tool",
					  "../tool_runner?tool_id=hb_g_suite_single_value_analysis_per_track_tool",
					  "../tool_runner?tool_id=hb_determine_suite_tracks_coinciding_with_another_suite"];
			subs9 = [1,3];
			llength9 = [15,19,21];
			
			createBox(ctx,490,20,240,text9,links9,subs9,llength9,"#FFFFE0");
	
			text10 = ["Statistical analysis of a GSuite",
					 "Similarity-based Analysis of Tracks in GSuite", 
					 "Determine repre- sentative and atypical tracks in a GSuite",
					 "Determine GSuite tracks coinciding with a target track",
					 "Determine coinciding track combinations from two suites",
					 "Distribution of Suite Data Along the Genome",
					 "Determine regions where GSuite tracks are enriched",
					 "Determine regions where GSuite tracks co-occur more strongly"];
			links10 = ["","","../tool_runner?tool_id=hb_g_suite_representative_and_untypical_track_tool",
					  "../tool_runner?tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool",
					  "../tool_runner?tool_id=hb_coinciding_tracks_from_two_g_suites_tool",
					  "","../tool_runner?tool_id=hb_multi_track_count_overrepresentation_bins",
					  "../tool_runner?tool_id=hb_multi_track_cooccurrence_bins"];
			subs10 = [1,5];		
			llength10 = [15,19,21];
			
			createBox(ctx,630,20,220,text10,links10,subs10,llength10,"#FFFFE0");
	
			text11 = ["Visual analysis of GSuites",
					 "Color-coded segments from a query track based on overlap with reference tracks in a GSuite", 
					 "Generate a heatmap based on overlap of segments from a query track with reference tracks in a GSuite",
					 "Generate a Venn diagram of overlap between tracks in a GSuite"];
			links11 = ["","../tool_runner?tool_id=hb_segments_overlap_visualization_tool",
					  "../tool_runner?tool_id=hb_segments_overlap_visualization_heatmap_tool",
					  "../tool_runner?tool_id=hb_create_bps_venn_d_iagram"];
			subs11 = [];
			llength11 = [15,19,20];
			
			createBox(ctx,490,470,175,text11,links11,subs11,llength11,"#FFFFE0");
	
			text12 = ["Domain- specific analysis ",
					 "Scan targets of a transcription factor", 
					 "Scan transcription factors of a genomic region"];
			links12 = ["","../tool_runner?tool_id=hb_all_targets_of_tfs",
					  "hyper?mako=generictool&tool_id=hb_all_tfs_of_regions"];
			subs12 = [];
			llength12 = [15,19,21];
			
			createBox(ctx,630,455,85,text12,links12,subs12,llength12,"#FFFFE0");
	
			text13 = ["Auxiliary tools",
					 "Generate the union track of all tracks in a GSuite"];
			links13 = ["","","../tool_runner?tool_id=hb_g_suite_download_files"];
			subs13 = [];
			llength13 = [15,19,21];
			
			createBox(ctx,630,645,65,text13,links13,subs13,llength13,"#FFFFE0");
			
			//1-7
			arrow(ctx,130,40,325,40);
			
			//1-4
			arrow(ctx,130,125,160,125);
			
			//2-4
			arrow(ctx,130,250,160,250);
	
			//2-7
			arrow(ctx,130,370,325,370);
	
			//4-5
			arrow(ctx,130,440,170,470);
	
			//4-8
			arrow(ctx,250,280,340,465);
			arrow(ctx,340,465,250,280);
	
			//3-4
			arrow(ctx,125,530,200,280);
	
			//3-END STAGE
			arrow(ctx,130,715,475,715);
			//3-8
			arrow(ctx,390,715,390,600);
	
			//8-5
			arrow(ctx,330,540,295,540);
	
			//6-8
			arrow(ctx,390,465,390,425);
			arrow(ctx,390,425,390,465);
	
			//7-END STAGE
			arrow(ctx,450,370,475,370);
	
			//4-6
			arrow(ctx,325,100,295,100);
			arrow(ctx,295,100,325,100);
	
			//4-7
			arrow(ctx,295,210,370,305);
			arrow(ctx,370,305,295,210);
	
			//4-END STAGE
			arrow(ctx,290,170,475,170);
	
			//7-5
			arrow(ctx,370,132,230,465);
	
			//6-7
			arrow(ctx,410,132,410,305);
	
			//5-7
			arrow(ctx,290,480,335,420);
	
	
			canvasListening(canvas);
			}
		}
		
		var JQNoCon = jQuery.noConflict();
		
		JQNoCon(document).ready(function() {
			draw();
		});
	   </script>
</head>


<body>


<div class="tabs">
    <ul class="tab-links" id="tab-links">
        <li class="active" value="tab1"><a href="#tab1">Main page</a></li>
        <li value="tab2"><a href="#tab2">Basic mode</a></li>
        <li value="tab3"><a href="#tab3">Advanced mode</a></li>
        <li value="tab4"><a href="#tab4">Examples</a></li>
        <li value="tab5"><a href="#tab5">About</a></li>
    </ul>

    <div class="tab-content">
        <div id="tab1" class="tab active">
		<div class="document">
		    <div class="main">
		    <h1>Welcome to the GSuite HyperBrowser</h1>
		    <h2>Manage and Analyze collections of genome-wide datasets</h2>
		    <p>
			   <!--older version <b>The GSuite HyperBrowser</b> provides functionality for managing and analyzing collections of
				genome-wide datasets (genomic tracks). At the core of the system is GSuite, a simple tabular format for
				representing genomic track collections (one genomic track per line) along with related metadata.-->
				<b>The GSuite HyperBrowser</b> system includes a range of tools to handle acquisition, processing and analysis 
				of collections of genomic tracks, represented in a simple tabular format, GSuite. Please proceed in either basic or advanced mode.
		    </p>

				<p style="infomessagesmall">
		    	<b>**IMPORTANT:</b> Please note that the GSuite HyperBrowser is planned to be published as part of
				<a href="https://hyperbrowser.uio.no" target="_blank"> the main HyperBrowser</a> server by January 2017. While you can use
				the current pre-published instance as you will, keep in mind that datasets and analyses histories created in this instance
					will not be accessible in the official published instance.
				</p>
		    <!--<p>
		    Two different operation modes are available to guide you through your analysis.
				Depending on your experience with the Genomic HyperBrowser (and Galaxy-based web tools in general),
				please select either Basic or Advanced user mode.
		    </p>-->
		    </div>
		    <div class="basic" id="basic" onclick="javascript:showDiv('tab1', 'tab2');" onmouseover="this.style.background='#F8F0B1';" onmouseout="this.style.background='#FBDF6E';" style="float:left;width:48%;overflow:hidden;margin-top:15px;margin-right:5px;">
		    <h2>Basic user mode ></h2>
		    <p style="height:100px;">
		    The Basic mode will let you select from pre-defined genome analysis questions and lead you
				through the corresponding simplified workflow. 
				<br><br>
				<span style="padding-left:10px;"><i>You will be provided with relevant example datasets
				necessary for running an analysis with the default parameter setup.</i></span>
				<!--Choose this mode if you are new to the GSuite HyperBrowser as well as the Genomic HyperBrowser.-->
		    </p>
		    </div>
		    <div class="advanced" id="advanced" onclick="javascript:showDiv('tab1', 'tab3');" onmouseover="this.style.background='#F8F0B1';" onmouseout="this.style.background='#FBDF6E';" style="overflow:hidden;margin-top:15px;">
		    <h2>Advanced user mode ></h2>
		    <p style="height:100px;">
		    The Advanced mode gives you full control of all analysis workflows and their intermediate steps.
				<br><br>
				<span style="padding-left:10px;"><i>You will be able to edit the content of GSuite 
				files and modify the full range of analysis parameters.</i></span>
				<!--Choose this mode if you have previous experience with the GSuite HyperBrowser
				or the Genomic HyperBrowser.-->
			<br><br>
		    </p>
		    </div>
		    <div id="screencast" class="advanced" style="overflow:auto;padding-bottom:10px;margin-top:15px;">
			<h2>Brief demos of the system (Screencasts)</h2>
			<p>
				Get familiar with the system through the short video guides:
			
			    <!-- <ul>
				<li><a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/16.08/104267/System_Overview_-_20160816_152607_39.html">
				GSuite HB overview</a></li>
				<li><a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/17.08/194005/example_analysis_final_imovie_-_20160817_180630_39.html">
				An example analysis (Celiac disease SNPs vs. DNase HSS)</a></li>
			     </ul>-->
			</p>
			<div class="video-container" style="width:49%;float:left;border:1px dotted black;box-sizing:border-box;padding-top:15px">
				<a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/16.08/104267/System_Overview_-_20160816_152607_39.html" target="_blank">
					<img src="hyperbrowser/images/videos/gsuite_overview.jpg" class="center" style="width:90%;height:auto;border:1px solid black;padding:0px"/>
				</a>
				<p class="center" style="height:32px">
					<a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/16.08/104267/System_Overview_-_20160816_152607_39.html" target="_blank">
                                	GSuite HyperBrowser overview</a>
				</p>
			</div>
			<div class="video-container" style="width:49%;float:right;border:1px dotted black;box-sizing:border-box;padding-top:15px;">
				<a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/17.08/194005/example_analysis_final_imovie_-_20160817_180630_39.html" target="_blank">
					<img src="hyperbrowser/images/videos/gsuite_example1.jpg" class="center" style="width:90%;height:auto;border:1px solid black;padding:0px;"/>
				</a>
				<p class="center" style="height:32px">
					<a href="https://screencast.uninett.no/relay/ansatt/geirksauio.no/2016/17.08/194005/example_analysis_final_imovie_-_20160817_180630_39.html" target="_blank">
					An example analysis (Celiac disease SNPs vs. DNase HSS)</a>
				</p>
			</div>
		    </div>

		</div>

	</div>

	<div id="tab2" class="tab">



	    <div class="document">
		<div class="main">
		    <h2>The GSuite HyperBrowser - Basic User Mode</h2>
		    <h3>
		    <span title="Hover mouse over words in bold if you are uncertain about their meaning">
		    <b>What is the biological question you would like to ask?</b>
		    </span>
		    </h3>
		    <p>(Tip: Click to expand the available biological questions in the scenario of interest.)</p>

<!-- 		    <p>
		    <b>**IMPORTANT:</b> Currently only sections (biological questions) marked with ** are functioning.
		    This is because we are still in a testing phase. For now use only the marked questions to test the functionality.
		    The rest of the sections will be filled out later. Suggestions for new sections are also welcomed.
		    </p> -->
		</div>

        <div class="question-wrapper" onclick="toggleDiv('#q1', '#i1');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <p>
                <img data-open='false' id="i1" src="hyperbrowser/images/div/DNAlying.png"
                width="20" height="20" />
                <span> - Explore disease-associated variation (GWAS SNPs)
 </span>
             </p>

        </div>
        <div id='q1' class="answer-wrapper">
        
            <div class="question" id="q11" onclick="toggleDiv('#a11', '#i11');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i11" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied trait-associated variants particularly active in certain cell types?

            </div>

        
        <div class="answer" id="a11">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing to what extent variants from a given query set fall into the active chromatin regions (determined by DNase-Seq, FAIRE-Seq or histone modification ChIP-Seq) of selected cell types. Any number of cell types can be uniformly investigated within a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox1_1')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_1" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific SNP variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the areas of active chromatin, with each dataset describing the genomic regions of DNaseI accessibility for a separate cell type).
<br>

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_1">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL01TX3JlZ2lvbnNfZXhwYW5kZWRfMTBrYi5iZWQ=&datatype=bed&job_name=sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions&basicQuestionId=1_1">sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_1">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=1_1">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each representing the active chromatin regions of a separate cell type, in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL0ROYXNlX2luXzQwX2NlbGxfdHlwZXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite of DNaseI accessibility for different cell types&basicQuestionId=1_1">sample GSuite of DNaseI accessibility for different cell types</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import active chromatin data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_1">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import active chromatin data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_1">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded cell type specific files can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_1">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_1">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=1_1">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q12" onclick="toggleDiv('#a12', '#i12');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i12" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied trait-associated variants preferentially located in certain chromatin states (in a given cell type)?

            </div>

        
        <div class="answer" id="a12">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment of query variants in regions associated with a particular chromatin state (determined by histone modification ChIP-Seq or computationally) in a cell type of interest. Any number of chromatin states can be uniformly investigated within a single analysis run. Various information, including genomic locations of individual histone modifications or their combinations, can be used to denote a chromatin state.
</div></br><img onclick="toggleInfoBox('#infoBox1_2')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_2" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific SNP variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the areas of various chromatin states, with each dataset describing the genomic regions associated with a separate chromatin state of a given cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_2">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL01TX3JlZ2lvbnNfZXhwYW5kZWRfMTBrYi5iZWQ=&datatype=bed&job_name=sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions&basicQuestionId=1_2">sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_2">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=1_2">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets related to a given cell type, each dataset representing the regions of a separate chromatin state; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2hpc3RfazU2Mi5nc3VpdGU=&datatype=gsuite&job_name=sample GSuite collection: genomic locations of various histone modifications for the K562 chronic myelogenous leukemia cell line&basicQuestionId=1_2">sample GSuite collection: genomic locations of various histone modifications for the K562 chronic myelogenous leukemia cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import chromatin state data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_2">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import chromatin state data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_2">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific chromatin states can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_2">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_2">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=1_2">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q13" onclick="toggleDiv('#a13', '#i13');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i13" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied trait-associated variants potentially disrupting the regulatory function of a given transcription factor (in certain cell types)?

            </div>

        
        <div class="answer" id="a13">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment of query variants within the binding sites of a particular transcription factor (TF). Disruption of binding sites (whether these were determined by TF ChIP-Seq, DamID or computational predictions based on binding motifs) of the selected TF can be uniformly investigated for any number of cell types in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox1_3')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_3" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific SNP variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of TF binding sites, with each dataset representing the binding sites of a given TF in a separate cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_3">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL01TX3JlZ2lvbnNfZXhwYW5kZWRfMTBrYi5iZWQ=&datatype=bed&job_name=sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions&basicQuestionId=1_3">sample track with Multiple Sclerosis-associated regions, expanded 10kb in both directions</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=1_3">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=1_3">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets related to a given TF, each dataset representing the binding sites of this TF in a separate cell type; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2NNeWNfYWNyb3NzXzQzX2NlbGxfdHlwZXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of the transcription factor c-myb in various cell types&basicQuestionId=1_3">sample GSuite collection: genomic locations of binding sites of the transcription factor c-myb in various cell types</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_3">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_3">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific cell types can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_3">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_3">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=1_3">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q14" onclick="toggleDiv('#a14', '#i14');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i14" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How similar are the supplied traits in terms of sharing their lead SNPs?

            </div>

        
        <div class="answer" id="a14">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the degree to which annotated lead SNPs are shared across selected traits. Any number of traits (each represented by a corresponding set of variants) can be included in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox1_4')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_4" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How do the individual tracks of a single suite overlap each other?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of trait-associated SNP variants, with each dataset representing a separate trait).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the variants associated with a separate trait; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2d3YXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of lead SNP variants for various traits&basicQuestionId=1_4">sample GSuite collection: genomic locations of lead SNP variants for various traits</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import trait-associated SNP data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_4">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import trait-associated SNP data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_4">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific traits can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_4">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_4">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing one of the following tools:
</li><ol type="a"><li>				- the <<EL|href===../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=1_4|title=== Are certain tracks of one suite coinciding particularly strongly
        with certain tracks of another suite?>> tool provides a selection of multiple methods for calculating a pair-wise overlap table for all the tracks of the suite;
</li><li>				- the <a href="../hyper?mako=generictool&tool_id=hb_pilot_page_overlap_between_tracks_tool&bmQid=1_4">Summary statistics per track in a GSuite</a> tool calculates overall overlap statistics (e.g. a table showing how many SNPs are shared among 1, 2, …, n tracks of the suite).
</li>
</ol>
</ol>
</div>
            </div>


        
            <div class="question" id="q15" onclick="toggleDiv('#a15', '#i15');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i15" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How do the supplied traits cluster based on the location of their lead SNPs?

            </div>

        
        <div class="answer" id="a15">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis exploring the clustering tendencies of selected traits. Any number of traits (each represented by a corresponding set of variants) can be included in a single analysis run. One of multiple measures, each of which assesses the distribution of individual traits’ associated variants along the genome, can be used as a basis for the clustering.
</div></br><img onclick="toggleInfoBox('#infoBox1_5')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_5" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How do the individual tracks of a single suite form clusters?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of trait-associated SNP variants, with each dataset representing a separate trait).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the variants associated with a separate trait; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2d3YXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of lead SNP variants for various traits&basicQuestionId=1_5">sample GSuite collection: genomic locations of lead SNP variants for various traits</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import trait-associated SNP data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_5">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import trait-associated SNP data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_5">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific traits can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_5">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_5">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing the <a href="../hyper?mako=generictool&tool_id=hb_clus_track_tool&isBasic=1&isBasic=True">ClusTrack: Cluster tracks in GSuite based on genome level similarity</a> tool.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q16" onclick="toggleDiv('#a16', '#i16');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i16" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Which of the supplied traits is represented by the most (or the least) unique lead SNP set?

            </div>

        
        <div class="answer" id="a16">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis calculating the relative ranking of each submitted trait based on its average enrichment ratio against the other supplied traits. Any number of traits (each represented by a corresponding set of variants) can be included in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox1_6')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox1_6" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "Given a suite of tracks, which tracks are the most or the least unique ones relative to the other tracks in the suite?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of trait-associated SNP variants, with each dataset representing a separate trait).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the variants associated with a separate trait; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2d3YXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of lead SNP variants for various traits&basicQuestionId=1_6">sample GSuite collection: genomic locations of lead SNP variants for various traits</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import trait-associated SNP data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=1_6">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import trait-associated SNP data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=1_6">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific traits can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=1_6">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=1_6">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing the <a href="../hyper?mako=generictool&tool_id=hb_pilot_page_similarity_and_uniqueness_of_tracks_tool&bmQid=1_6">Similarity and uniqueness of tracks</a> tool.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q17" onclick="toggleDiv('#a17', '#i17');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i17" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- What I am interested in does not match any of the suggestions above..

            </div>

        
        <div class="answer" id="a17">
            The GSuite HyperBrowser is a generic system for analyzing suites of tracks, and as such is not in any way restricted to the specific example scenarios described above.
<br>
<br>If you are interested in a question that does not directly match any of the examples above, you might still get an idea of how to perform your analysis of interest by reading through the examples above and looking for parallels to your question.
<br>
<br>Alternatively, you can switch to <a href="javascript:;"  onclick="showDiv('tab1', 'tab3’);">advanced mode</a>, and look at the general descriptions of the analytical options provided by the GSuite HyperBrowser.
<br>
<br>Does any of the following generic tools seem suitable for the problem at hand?
<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_representative_and_untypical_track_tool&isBasic=1&isBasic=True&bmQid=1_7">Which tracks (in a suite) are most representative and most atypical?</a> - intended for analyzing single collections.
<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=1_7">Which tracks (in a suite) coincide most strongly with a separate single track?</a> - intended for analysis of a single track of interest against a collection of tracks.
<br><<EL|href===../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=1_7|title=== Are certain tracks of one suite coinciding particularly strongly
        with certain tracks of another suite?>> - intended for analyzing two collections of tracks against each other.
<br>
<br>You are also more than welcome to contact us at <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@sgub-resworbrepyh</a>. We have provided help for a broad range of investigations before, and are happy to provide noncommittal advice as well as take part in collaborations.
<br>

            </div>


        
</div>
        <div class="question-wrapper" onclick="toggleDiv('#q2', '#i2');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <p>
                <img data-open='false' id="i2" src="hyperbrowser/images/div/DNAlying.png"
                width="20" height="20" />
                <span> - Explore cancer mutations (somatic variation)
 </span>
             </p>

        </div>
        <div id='q2' class="answer-wrapper">
        
            <div class="question" id="q21" onclick="toggleDiv('#a21', '#i21');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i21" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) particularly active in certain cell types?

            </div>

        
        <div class="answer" id="a21">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing to what extent somatic variants from a given query set fall into the active chromatin regions (determined by DNase-Seq, FAIRE-Seq or histone modification ChIP-Seq) of selected cell types. Any number of cell types can be uniformly investigated within a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox2_1')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox2_1" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific somatic variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the areas of active chromatin, with each dataset describing the genomic regions of DNaseI accessibility for a separate cell type).
<br>

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_1">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gtrack&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL1RDR0FfUFJBRC5ndHJhY2s=&datatype=gtrack&job_name=sample track with genomic locations of somatic variants in prostate adenocarcinoma (the COAD set from The Cancer Genome Atlas)&basicQuestionId=2_1">sample track with genomic locations of somatic variants in prostate adenocarcinoma (the COAD set from The Cancer Genome Atlas)</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_1">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=2_1">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each representing the active chromatin regions of a separate cell type; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL0ROYXNlX2luXzQwX2NlbGxfdHlwZXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite of DNaseI accessibility for different cell types&basicQuestionId=2_1">sample GSuite of DNaseI accessibility for different cell types</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import active chromatin data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=2_1">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import active chromatin data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=2_1">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded cell type specific files can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=2_1">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=2_1">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=2_1">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q22" onclick="toggleDiv('#a22', '#i22');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i22" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) preferentially located in certain chromatin states (in a given cell type)?

            </div>

        
        <div class="answer" id="a22">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment of query somatic variants in regions associated with a particular chromatin state (determined by histone modification ChIP-Seq or computationally) in a cell type of interests. Any number of chromatin states can be uniformly investigated within a single analysis run. Various information, including genomic locations of individual histone modifications or their combinations, can be used to denote a chromatin state.
</div></br><img onclick="toggleInfoBox('#infoBox2_2')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox2_2" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific somatic variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the areas of various chromatin states, with each dataset describing the genomic regions associated with a separate chromatin state of a given cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_2">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gtrack&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL1RDR0FfUFJBRC5ndHJhY2s=&datatype=gtrack&job_name=sample track with genomic locations of somatic variants in prostate adenocarcinoma (the COAD set from The Cancer Genome Atlas)&basicQuestionId=2_2">sample track with genomic locations of somatic variants in prostate adenocarcinoma (the COAD set from The Cancer Genome Atlas)</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_2">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=2_2">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets related to a given cell type, each dataset representing the regions of a separate chromatin state; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2hpc3RfazU2Mi5nc3VpdGU=&datatype=gsuite&job_name=sample GSuite collection: genomic locations of various histone modifications for the K562 chronic myelogenous leukemia cell line&basicQuestionId=2_2">sample GSuite collection: genomic locations of various histone modifications for the K562 chronic myelogenous leukemia cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import chromatin state data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=2_2">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import chromatin state data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=2_2">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific chromatin states can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=2_2">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=2_2">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=2_2">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q23" onclick="toggleDiv('#a23', '#i23');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i23" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) potentially disrupting the regulatory function of certain transcription factors (in a given cell type)? (Example of a specialized tool.)

            </div>

        
        <div class="answer" id="a23">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment and impact of query somatic variants within the binding sites of selected transcription factors (TFs) in a cell type of interest. For every case of a binding site co-location with a variant, the change in affected motif’s ability to bind its associated TF is evaluated (both increases and decreases in binding probability are being reported). Any number of TFs, represented by their binding sites (determined by TF ChIP-Seq, DamID or computational predictions based on binding motifs), can be uniformly investigated within a single analysis run.
<br>
</div></br><img onclick="toggleInfoBox('#infoBox2_3')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox2_3" class="infoBox">
	This example task can be seen as a special instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the genomic locations of specific somatic variants);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of TF binding sites, with each dataset representing the binding sites of a separate TF in a given cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the variants, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_3">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gtrack&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fdHJhY2tfbGljYV9jbi5ndHJhY2s=&datatype=gtrack&job_name=sample track with genomic locations of somatic variants in liver cancer (the LICA-CN set from The Cancer Genome Atlas)&basicQuestionId=2_3">sample track with genomic locations of somatic variants in liver cancer (the LICA-CN set from The Cancer Genome Atlas)</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=2_3">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=2_3">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets related to a given cell type, each dataset representing the binding sites of a separate TF; in GSuite format, with a metadata column named "pwm" that for each TF lists Transfac PWM IDs, which should be used in the TF binding change analyses). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fZ3N1aXRlX3Rmc193aXRoX3B3bXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of selected TFs in K562 (CTCF, c-Jun, c-Myc, GATA-1 and more) with added PWM metadata&basicQuestionId=2_3">sample GSuite collection: genomic locations of binding sites of selected TFs in K562 (CTCF, c-Jun, c-Myc, GATA-1 and more) with added PWM metadata</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=2_3">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a> (the <a href="../hyper?mako=generictool&tool_id=hb_match_tf_with_pwm&isBasic=1&isBasic=True&bmQid=2_3">Match tf with pwm</a> tool can be used for automatically attaching a well-formed "pwm" column to the resulting GSuite);
</li><li>				- (recommended for advanced users only) import TF binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=2_3">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific TFs can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=2_3">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=2_3">Compile GSuite from archive (Zip/tar)</a> tool); the <a href="../hyper?mako=generictool&tool_id=hb_match_tf_with_pwm&isBasic=1&isBasic=True&bmQid=2_3">Match tf with pwm</a> tool can be used for automatically attaching a well-formed "pwm" column to the resulting GSuite (this column can be further manually edited with the <a href="../hyper?mako=generictool&tool_id=hb_edit_gsuite_metadata_tool&isBasic=1&isBasic=True&basicQuestionId=2_3">Edit a metadata column in a GSuite</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_tf_binding_disruption">TF binding disruption</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q24" onclick="toggleDiv('#a24', '#i24');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i24" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Do the supplied somatic variants occur preferentially in certain genes?

            </div>

        
        <div class="answer" id="a24">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing absolute and patient-wise somatic mutation rates within individual genes. A normalization factor is calculated for each gene based on the cumulative length of genomic regions assigned to it (in case of targeted assays, only the targeted regions should be taken into account). Multiple patients (each corresponding to a set of somatic variants) and genes (each represented by a set of genomic regions, e.g., exons) can be uniformly investigated in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox2_4')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox2_4" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How are the track elements of one suite distributed among the tracks of another suite?” Two input collections, each being a separate GSuite instance, are required:
<br>A dataset of cases: independent sets of genomic events/features (e.g. multiple patients, each with an own list of genomic locations that represent somatic variants);
<br>A collection of potential targets: datasets of closely related genomic features (e.g. a collection of datasets which map the genomic locations of exons, with each dataset corresponding to one gene).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of patient tracks (a collection of genomic location datasets, each dataset representing the somatic variants of a separate patient; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fZ3N1aXRlX3N2X2NvbGFkZS5nc3VpdGU=&datatype=gsuite&job_name=sample GSuite collection: somatic variant locations for 216 Colon adenocarcinoma patients (the COAD dataset from The Cancer Genome Atlas)&basicQuestionId=2_4">sample GSuite collection: somatic variant locations for 216 Colon adenocarcinoma patients (the COAD dataset from The Cancer Genome Atlas)</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import somatic variation data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=2_4">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a> (please note that additional post-processing will be necessary for creating a patient-wise collection);
</li><li>				- (recommended for advanced users only) upload your own patient-wise somatic variation data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a>); individually uploaded files representing specific patients can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=2_4">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=2_4">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Fetch the suite of gene tracks (a collection of genomic location datasets, each dataset representing the exons of a separate gene; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fZ3N1aXRlX3RjZ2FfZXhvbWUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: exon locations for 560 genes included in the Cancer Census&basicQuestionId=2_4">sample GSuite collection: exon locations for 560 genes included in the Cancer Census</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- (recommended for advanced users only) upload your own gene-region data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a>); individually uploaded files representing specific genes can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=2_4">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=2_4">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the data by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_determine_suite_tracks_coinciding_with_another_suite&isBasic=1&isBasic=True&bmQid=2_4">Determine suite tracks coinciding with another suite</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q25" onclick="toggleDiv('#a25', '#i25');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i25" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- What I am interested in does not match any of the suggestions above..

            </div>

        
        <div class="answer" id="a25">
            The GSuite HyperBrowser is a generic system for analyzing suites of tracks, and as such is not in any way restricted to the specific example scenarios described above.
<br>
<br>If you are interested in a question that does not directly match any of the examples above, you might still get an idea of how to perform your analysis of interest by reading through the examples above and looking for parallels to your question.
<br>
<br>Alternatively, you can switch to <a href="javascript:;"  onclick="showDiv('tab1', 'tab3’);">advanced mode</a>, and look at the general descriptions of the analytical options provided by the GSuite HyperBrowser.
<br>
<br>Does any of the following generic tools seem suitable for the problem at hand?
<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_representative_and_untypical_track_tool&isBasic=1&isBasic=True&bmQid=2_5">Which tracks (in a suite) are most representative and most atypical?</a> - intended for analyzing single collections.
<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=2_5">Which tracks (in a suite) coincide most strongly with a separate single track?</a> - intended for analysis of a single track of interest against a collection of tracks.
<br><<EL|href===../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=2_5|title=== Are certain tracks of one suite coinciding particularly strongly
        with certain tracks of another suite?>> - intended for analyzing two collections of tracks against each other.
<br>
<br>You are also more than welcome to contact us at <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@sgub-resworbrepyh</a>. We have provided help for a broad range of investigations before, and are happy to provide noncommittal advice as well as take part in collaborations.
<br>

            </div>


        
</div>
        <div class="question-wrapper" onclick="toggleDiv('#q3', '#i3');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <p>
                <img data-open='false' id="i3" src="hyperbrowser/images/div/DNAlying.png"
                width="20" height="20" />
                <span> - Explore transcription factor binding
 </span>
             </p>

        </div>
        <div id='q3' class="answer-wrapper">
        
            <div class="question" id="q31" onclick="toggleDiv('#a31', '#i31');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i31" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How do the supplied transcription factors cluster based on the location of their binding sites (i.e. based on the targets of their regulatory activity)?

            </div>

        
        <div class="answer" id="a31">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis exploring the clustering tendencies of selected transcription factors (TFs). Any number of TFs (each represented by a corresponding set of binding sites) can be included in a single analysis run. One of multiple measures, each of which assesses the distribution of individual TFs’ associated binding sites along the genome, can be used as a basis for the clustering.
</div></br><img onclick="toggleInfoBox('#infoBox3_1')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_1" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How do the individual tracks of a single suite form clusters?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of transcription factor (TF)-associated binding sites, with each dataset representing a separate TF).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the binding sites associated with a separate TF; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlLzE1OV9URnNfaW5fZ20xMjg3OF9jZWxsX3R5cGUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line&basicQuestionId=3_1">sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF-associated binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_1">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF-associated binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_1">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific TFs can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_1">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_1">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing the <a href="../hyper?mako=generictool&tool_id=hb_clus_track_tool&isBasic=1&isBasic=True">ClusTrack: Cluster tracks in GSuite based on genome level similarity</a> tool.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q32" onclick="toggleDiv('#a32', '#i32');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i32" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How much does binding (i.e. the set of supplied binding sites) of a given transcription factor vary across cell types?

            </div>

        
        <div class="answer" id="a32">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the degree to which binding sites of a given transcription factor (TF) vary across cell types. Any number of cell types, each represented by a corresponding set of binding sites for the selected TF, can be uniformly investigated in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox3_2')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_2" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How do the individual tracks of a single suite overlap each other?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of TF binding sites, with each dataset representing the binding sites of a given TF in a separate cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets related to a given TF, each dataset representing the binding sites of this TF in a separate cell type; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2NNeWNfYWNyb3NzXzQzX2NlbGxfdHlwZXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of the transcription factor c-myb in various cell types&basicQuestionId=3_2">sample GSuite collection: genomic locations of binding sites of the transcription factor c-myb in various cell types</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_2">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_2">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific cell types can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_2">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_2">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the suite of tracks by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_pilot_page_overlap_between_tracks_tool&bmQid=3_2">Summary statistics per track in a GSuite</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q33" onclick="toggleDiv('#a33', '#i33');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i33" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How similar are the supplied transcription factors in terms of sharing their binding sites?

            </div>

        
        <div class="answer" id="a33">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the degree to which binding sites are shared across selected TFs. Any number of TFs (each represented by a corresponding set of binding sites) can be included in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox3_3')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_3" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How do the individual tracks of a single suite overlap each other?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of transcription factor (TF) binding sites, with each dataset representing a separate TF).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the binding sites associated with a separate TF; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlLzE1OV9URnNfaW5fZ20xMjg3OF9jZWxsX3R5cGUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line&basicQuestionId=3_3">sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF-associated binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_3">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF-associated binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_3">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific TFs can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_3">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_3">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing one of the following tools:
</li><ol type="a"><li>				- the <<EL|href===../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=3_3|title=== Are certain tracks of one suite coinciding particularly strongly
        with certain tracks of another suite?>> tool provides a selection of multiple methods for calculating a pair-wise overlap table for all the tracks of the suite;
</li><li>				- the <a href="../hyper?mako=generictool&tool_id=hb_pilot_page_overlap_between_tracks_tool&bmQid=3_3">Summary statistics per track in a GSuite</a> tool calculates overall overlap statistics (e.g. a table showing how many binding sites are shared among 1, 2, …, n TF tracks of the suite).
</li>
</ol>
</ol>
</div>
            </div>


        
            <div class="question" id="q34" onclick="toggleDiv('#a34', '#i34');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i34" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Which of the supplied transcription factors is represented by the most (or the least) unique binding site set?

            </div>

        
        <div class="answer" id="a34">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis calculating the relative ranking of each submitted transcription factor (TF) based on its average enrichment ratio against the other supplied TFs. Any number of TFs (each represented by a corresponding set of binding sites) can be included in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox3_4')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_4" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "Given a suite of tracks, which tracks are the most or the least unique ones relative to the other tracks in the suite?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of transcription factor (TF)-associated binding sites, with each dataset representing a separate TF).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the binding sites associated with a separate TF; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlLzE1OV9URnNfaW5fZ20xMjg3OF9jZWxsX3R5cGUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line&basicQuestionId=3_4">sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import trait-associated SNP data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_4">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF-associated binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_4">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific TFs can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_4">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_4">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the tracks of the suite by utilizing the <a href="../hyper?mako=generictool&tool_id=hb_pilot_page_similarity_and_uniqueness_of_tracks_tool&bmQid=3_4">Similarity and uniqueness of tracks</a> tool.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q35" onclick="toggleDiv('#a35', '#i35');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i35" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- In which cell types do the supplied transcription factor binding sites fall into regulatory active regions?

            </div>

        
        <div class="answer" id="a35">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment of supplied transcription factor binding sites (for a specific transcription factor) in regulatory active regions of multiple cell types. Locations of various genomic features (including DNaseI accessibility sites, FAIRE sites or specific histone modifications, e.g. H3K4me1 and H3K27ac) can be used as a proxy for regulatory active regions. Any number of cell types (or cell type - feature combinations) can be uniformly investigated in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox3_5')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_5" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the binding sites of a given transcription factor);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the areas of active chromatin, with each dataset describing the genomic regions of DNaseI accessibility for a separate cell type).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the binding sites of a given transcription factor, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_5">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlLzE1OV9URnNfaW5fZ20xMjg3OF9jZWxsX3R5cGUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line&basicQuestionId=3_5">sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_5">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=3_5">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the active chromatin regions of a separate cell type; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL0ROYXNlX2luXzQwX2NlbGxfdHlwZXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite of DNaseI accessibility for different cell types&basicQuestionId=3_5">sample GSuite of DNaseI accessibility for different cell types</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import active chromatin data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_5">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import active chromatin data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_5">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded cell type specific files can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_5">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_5">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=3_5">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q36" onclick="toggleDiv('#a36', '#i36');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i36" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- How do the supplied transcription factor binding sites overlap various traits’ associated DNA variants?

            </div>

        
        <div class="answer" id="a36">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis assessing the enrichment of trait-associated variants in the binding sites of a query transcription factor (TF). Any number of traits (each represented by its own set of associated variants, typically lead SNPs found through a GWAS) can be uniformly investigated in a single analysis run.
</div></br><img onclick="toggleInfoBox('#infoBox3_6')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_6" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "How does a query track overlap the different tracks of a suite?", where:
<br>The query track is a dataset of genomic features (e.g. the binding sites of a given transcription factor);
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the genomic locations of trait-associated SNP variants, with each dataset representing a separate trait).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track (a set of genomic locations representing the binding sites of a given transcription factor, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_6">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable query track available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlLzE1OV9URnNfaW5fZ20xMjg3OF9jZWxsX3R5cGUuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line&basicQuestionId=3_6">sample GSuite collection: genomic locations of binding sites of various TFs for the gm12878 lymphoblastoid cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_6">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=3_6">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the variants associated with a separate trait; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history;
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2d3YXMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of lead SNP variants for various traits&basicQuestionId=3_6">sample GSuite collection: genomic locations of lead SNP variants for various traits</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import trait-associated SNP data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_6">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import trait-associated SNP data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_6">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific traits can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_6">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_6">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Analyze the query track against the tracks of the suite by utilizing the tool <a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=3_6">Which tracks (in a suite) coincide most strongly with a separate single track?</a>.
</li>
</ol>
</div>
            </div>


        
            <div class="question" id="q37" onclick="toggleDiv('#a37', '#i37');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i37" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Which of the supplied genomic regions contain binding sites of a given transcription factor? (Example of a specialized tool.)

            </div>

        
        <div class="answer" id="a37">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis for determining the overlap between genomic regions of interest and the (putative) binding sites of a particular transcription factor (TF). Multiple sets of genomic regions can be uniformly investigated in a single analysis run. Input genomic regions with a TF binding site overlap are included in the output together with summary plots.
</div></br><img onclick="toggleInfoBox('#infoBox3_7')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_7" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "Which track elements from a query suite of tracks overlap the target genomic track?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the putative active enhancer regions of K562 cells, with each dataset describing the enhancer regions according to a separate definition/histone modification pattern);
<br>The target track is a single dataset of genomic features (e.g. the putative binding sites of a particular transcription factor).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the query track/suite of tracks (a collection of one or more genomic location datasets, each dataset representing regions which should be scanned for TF binding sites; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable track/GSuite file available in your current HyperBrowser history or if you plan to use data from the HyperBrowser repository (the tool utilized in the last step of this analysis offers a range of data collections for this purpose);
</li><li>				- click here to load a __dgs_k562_enhancers__ (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import enhancer region data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_7">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import enhancer region data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_7">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific enhancer sets can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_7">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_7">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Fetch the target track (a set of genomic locations representing TF binding sites, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_7">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable track file available in your current HyperBrowser history or if you plan to use data from the HyperBrowser repository (the tool utilized in the last step of this analysis offers a range of TF binding site datasets for this purpose);
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fdHJhY2tfbXljX2JzLmJlZA==&datatype=bed&job_name=sample track with Myc binding sites&basicQuestionId=3_7">sample track with Myc binding sites</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_7">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=3_7">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Utilize the tool <a href="../hyper?mako=generictool&tool_id=hb_all_targets_of_tfs&isBasic=1&isBasic=True&bmQid=3_7">Scan transcription factors of a genomic region</a> by
</li><ol type="a"><li>				- selecting a genomic track (or a suite of genomic tracks) representing the regions which should be scanned for TF binding sites - either from the current HyperBrowser history or from the pre-constructed list of suitable repository items;
</li><li>				- selecting a genomic track of TF binding sites - either from the current HyperBrowser history or from the pre-constructed list of suitable repository items;
</li><li>				- running the overlap analysis.
</li>
</ol>
</ol>
</div>
            </div>


        
            <div class="question" id="q38" onclick="toggleDiv('#a38', '#i38');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <img data-open='false' id="i38" src="hyperbrowser/images/div/DNAlying.png" width="15" height="15" />
            	- Which transcription factors have binding sites located within the given genomic region(s)? (Example of a specialized tool.)

            </div>

        
        <div class="answer" id="a38">
            <div id="analysisDetails">
<h3 class="welcomeAnalysisDetails">Analysis details</h3>
	An analysis for determining the overlap between genomic regions of interest and the (putative) binding sites of multiple transcription factors (TFs). A selected set of genomic regions can be uniformly scanned against multiple TF binding site datasets in a single analysis run. Input genomic regions with a TF binding site overlap are included in the output together with a TF co-occupancy summary.
</div></br><img onclick="toggleInfoBox('#infoBox3_8')" data-open="false" id="img" + "_" + str(self.questionCount) + " src="june_2007_style/HB/info_small.png" width="15" height="15" />
<div id="infoBox3_8" class="infoBox">
	This example task can be seen as a concrete instance of the more general question "Which tracks of a target suite of tracks overlap the query genomic track?", where:
<br>The suite of tracks (a GSuite instance) is a collection of related genomic feature datasets (e.g. a collection of datasets which map the putative transcription factor binding sites, with each dataset describing the binding sites of a separate transcription factor);
<br>The query track is a dataset of genomic features (e.g. putative active enhancer regions of K562 cells).

</div>
<div class="analysisSteps">
<h3 class="welcomeAnalysisSteps">Analysis steps</h3>
<ol>
<li>			- Fetch the suite of tracks (a collection of genomic location datasets, each dataset representing the binding sites of a separate TF; in GSuite format). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable GSuite file available in your current HyperBrowser history or if you plan to use data from the HyperBrowser repository (the tool utilized in the last step of this analysis offers a range of TF binding site datasets available for selection - a GSuite collection is automatically constructed from the selected datasets);
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_gsuite&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fZ3N1aXRlX3NlbGVjdGVkX3RmYnMuZ3N1aXRl&datatype=gsuite&job_name=sample GSuite collection: genomic locations of binding sites of selected TFs (CEBPB, FOS, JUN, MYC, NANOG, NR2F2, TEAD4)&basicQuestionId=3_8">sample GSuite collection: genomic locations of binding sites of selected TFs (CEBPB, FOS, JUN, MYC, NANOG, NR2F2, TEAD4)</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- import TF binding-site data from a public data source using the curated search tool <a href="../hyper?mako=generictool&tool_id=hb_track_global_search_tool&basicQuestionId=3_8">Create a GSuite from an integrated catalog of genomic datasets (track selection by data-type)</a>;
</li><li>				- (recommended for advanced users only) import TF binding-site data from a public data source (e.g. with the help of tool <a href="../hyper?mako=generictool&tool_id=hb_track_source_test_tool&basicQuestionId=3_8">Create a remote GSuite from a public repository (track selection by data-source)</a>) or upload your own data (e.g. by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool); individually uploaded files representing specific TFs can be combined into a single GSuite instance with the tool <a href="../hyper?mako=generictool&tool_id=hb_create_g_suite_file_from_history_elements_tool&basicQuestionId=3_8">Create a GSuite from datasets in your history</a> (alternatively, an uploaded zip/tar archive containing such files can serve as input of the <a href="../hyper?mako=generictool&tool_id=hb_compile_g_suite_from_archive_tool&basicQuestionId=3_8">Compile GSuite from archive (Zip/tar)</a> tool);
</li>
</ol>
<li>			- Fetch the query track (a set of genomic locations representing regions which should be scanned for TF binding sites, in BED, GTrack or some of the other <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_8">supported file formats</a>). You can:
</li><ol type="a"><li>				- skip this step if you already have a suitable track file available in your current HyperBrowser history or if you plan to use data from the HyperBrowser repository (the tool utilized in the last step of this analysis offers a range of tracks suitable for this purpose);
</li><li>				- click here to load a <a href="../tool_runner?tool_id=file_import_bed&dbkey=hg19&runtool_btn=yes&input=L3NvZnR3YXJlL2dhbGF4eS9nYWxheHlfZ3N1aXRlX3N1Ym1pdC9zdGF0aWMvaHlwZXJicm93c2VyL2RhdGEvZ3N1aXRlL2RlbW9fdHJhY2tfazU2Ml9lbmhhbmNlcnMuYmVk&datatype=bed&job_name=sample track with genomic locations of enhancer regions active within the K562 chronic myelogenous leukemia cell line&basicQuestionId=3_8">sample track with genomic locations of enhancer regions active within the K562 chronic myelogenous leukemia cell line</a> (you will be automatically redirected back to this page if you choose this option);
</li><li>				- upload your own file to the current HyperBrowser history by using the <a href="../tool_runner?tool_id=upload1">Upload file</a> tool (if the file is not in one of the <a href="welcome_formats.html?aaa=bbb&basicQuestionId=3_8">supported file formats</a>, you can convert it to the GTrack format by using the tool <a href="../hyper?mako=generictool&tool_id=hb_tabular_to_gtrack_tool&basicQuestionId=3_8">Create GTrack file from unstructured tabular data</a>);
</li>
</ol>
<li>			- Utilize the tool <a href="../hyper?mako=generictool&tool_id=hb_all_tfs_of_regions&isBasic=1&isBasic=True&bmQid=3_8">Scan transcription factors of a genomic region</a> by
</li><ol type="a"><li>				- selecting a suite of genomic tracks representing the TF binding sites - either from the current HyperBrowser history or from the pre-constructed list of suitable repository items;
</li><li>				- selecting a genomic track with regions which should be scanned for TF binding sites - either from the current HyperBrowser history or from the pre-constructed list of suitable repository items;
</li><li>				- running the overlap analysis.
</li>
</ol>
</ol>
</div>
            </div>


        
</div>
        <div class="question-wrapper" onclick="toggleDiv('#q4', '#i4');" onmouseover="this.style.background='#F8F0B1';this.style.border='1px solid #F8F0B1';"
            onmouseout="this.style.background='#FBDF6E';this.style.border='1px dotted #333333';">
            <p>
                <img data-open='false' id="i4" src="hyperbrowser/images/div/DNAlying.png"
                width="20" height="20" />
                <span> - What I am interested in does not match any of the suggestions above.
 </span>
             </p>

        </div>
        <div id='q4' class="answer-wrapper">
        		<div class="answer"  style="display:block">
<!--						The GSuite HyperBrowser is a generic system for analyzing suites of tracks, that is not in any way restricted to the specific example scenarios described above.
			<br>
			<br>If you are interested in a question that does not directly match any of the examples above, you might still get an idea of how to perform your analysis of interest by reading through the examples above and look for parallels to your question.
			<br>
			<br>Alternatively, you can switch to <a href="javascript:;"  onclick="showDiv('tab1', 'tab3â€™);â€�>advanced mode</a>, and look at the general descriptions of the analytical options provided by the GSuite HyperBrowser.
			<br>
			<br>You are also more than welcome to contact us in the HyperBrowser team at <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@sgub-resworbrepyh</a>. We have provided help to a broad range of investigations before, and are happy to provide noncommittal advice or also to take part in collaborations.
			<br>If not, these are the generic options, could any of them match?
			<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_representative_and_untypical_track_tool&isBasic=1&isBasic=True&bmQid=3_9">Determine representative and atypical tracks in a GSuite </a> - use it when you need to analyze a single collection
			<br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=3_9">Determine GSuite tracks coinciding with a separate track</a> - use it when you have a single track of interest that needs to be analyzed against a collection of tracks
			<br><a href="../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=3_9">Determine coinciding track combinations from two GSuites</a> - use it when you have two collections of tracks which you want to analyze against each other
			<br>-->
		            <br>	The GSuite HyperBrowser is a generic system for analyzing suites of tracks, and as such is not in any way restricted to the specific example scenarios described above.
                <br>
                <br>If you are interested in a question that does not directly match any of the examples above, you might still get an idea of how to perform your analysis of interest by reading through the examples above and looking for parallels to your question.
                <br>
                <br>Alternatively, you can switch to <a href="javascript:;"  onclick="showDiv('tab1', 'tab3â€™);">advanced mode</a>, and look at the general descriptions of the analytical options provided by the GSuite HyperBrowser.
                <br>
                <br>Does any of the following generic tools seem suitable for the problem at hand?
                <br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_representative_and_untypical_track_tool&isBasic=1&isBasic=True&bmQid=4_0">Determine representative and atypical tracks in a GSuite </a> - intended for analyzing single collections.
                <br><a href="../hyper?mako=generictool&tool_id=hb_g_suite_tracks_coinciding_with_query_track_tool&isBasic=1&isBasic=True&bmQid=4_0">Determine GSuite tracks coinciding with a separate track</a> - intended for analysis of a single track of interest against a collection of tracks.
                <br><a href="../hyper?mako=generictool&tool_id=hb_coinciding_tracks_from_two_g_suites_tool&isBasic=1&isBasic=True&bmQid=4_0">Determine coinciding track combinations from two GSuites</a> - intended for analyzing two collections of tracks against each other.
                <br>
                <br>You are also more than welcome to contact us at <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@sgub-resworbrepyh</a>. We have provided help for a broad range of investigations before, and are happy to provide noncommittal advice as well as take part in collaborations.
                <br>
		</div>

		</div>
		</div>
	</div>


	<div id="tab3" class="tab">

	    <div class="document">
		<div class="main">
		<div class="gsuitebox">
		    <h2 class="topheader">The GSuite HyperBrowser - Advanced User Mode</h2>
		    <p>
		    <b>The GSuite HyperBrowser</b> offers an easy and straightforward way to manage and analyze collections of
		    <a href="welcome_gentrack.html">genomic tracks</a>
		    (datasets of coordinates in a reference genome). At the core of the system is GSuite, a simple tabular format
		    for representing genomic track collections (one genomic track per line) along with related metadata.
		    Work with track collections typically requires three stages:
		    </p>
		    <ol type="1">
			    <li>
			    Compiling track collections (GSuites) from local files or external repositories.
			    </li>
			    <li>
			    Customizing track collections (GSuites) for analysis through textual manipulation and/or preprocessing.
			    </li>
			    <li>
			    Analyzing track collections (GSuites) - investigating the relations between tracks of a single collection or between collections.
			    </li>
		    </ol>

		    <p>
			    All GSuite tools are accessible from the left-hand "Tools" menu and via hyperlinks from the diagram displayed below. The diagram shows the individual tools grouped according to the workflow stages.
		    </p>

		    <h3 class="overview" align="center">GSuite Analysis Workflow Diagram</h3>
		    <h4 class="overview_subtitle" align="center">(follow the links in the diagram to access the tools directly)</h4>

		    <div style="width:900px; height:750px; margin: 0 auto; padding: 5px;">
			<canvas id="myCanvas" width="770" height="750">
			Your browser does not support the HTML5 canvas tag.
			</canvas>
			</div>

		    <p>
		    The power of the GSuite HyperBrowser is best seen when its tools are combined in a full investigation scenario. We provide several <a href="welcome.html?tabOn=tab4">examples</a>,
		    which illustrate the standard workflow structure, involving tools from the three main stages of a typical analysis. With annotated step-by-step analysis histories that include everything from the dataset acquisition to final results, these examples demonstrate solutions to a selection of complex research questions.
		    </p>
		    <p>
		    Please note that GSuite files come in three variants, referring to either
		    <a href="welcome_note.html">external (remote) datasets, local textual datasets or local binary datasets</a>.
		    Remote datasets need to be downloaded with the
		    <a href="../tool_runner?tool_id=hb_g_suite_download_files">Fetch tool</a>
		    before they can be further customized or analyzed, and textual datasets must be converted to binary format with the
		    <a href="../tool_runner?tool_id=hb_preprocess_g_suite_tracks_tool">Preprocessing tool</a>
		    before they can be used in an analysis.
		    </p>
		</div>
		</div>
	    </div>
	</div>

	<div id="tab4" class="tab">

	    <div class="document">
			<div class="main">
				<h2>
					Examples of analyses using GSuite
				</h2>

				<p>
					To demonstrate the purpose and capabilities of the GSuite HyperBrowser, we assembled a collection of complex example investigations. Our use-cases include three categories: exploration of genomic track characteristics, reproduction of a previously published study and answering of a novel research question. Each example is provided with an annotated analysis history, which documents the step-by-step process leading from initial dataset acquisition to final results while retaining all intermediate data files and parameter choices. Although each of the cases presents an objectively non-trivial task, both regarding data management and analysis methodology, the following self-contained GSuite-enabled solutions are very simple in terms of work required by the user.
				</p>
				<p>
					(The individual cases can be expanded/hid by a simple click.)
				</p>
			</div>



			<div class="tab4collectionTitle" id="T3">
				<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" height="20" width="20" id="B3" class='-off' /></div>
					- Chromatin accessibility (DHSs) across cell types
			</div>
			<div class="tab4collectionDesc" id="D3">

				<div class="show_hide" id="T31"><div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" id="B31" height="20" width="20"  class='-off' /></div> - Background</div>
				<div class="sliding" id="D31">

				<div class="tab4collectionDescDetails">Regulation of gene expression is governed by a range of proteins, which, in order to operate, require access to their respective target binding sites within the genome. Accessibility of a genomic location is determined by the local chromatin conformation, which can vary depending on cell type and conditions. Accessible sites allow not only potential interaction with sequence-specific DNA-binding proteins, but also with enzymes that can operate on any exposed DNA sequence, such as Deoxyribonuclease I (DNase I). A DNase I hypersensitive site (DHS) is an experimentally determined genomic location, where DNase I showed cleaving activity and thus demonstrated accessibility (indicating a potential for regulatory function). DHS profiles can therefore be used as non-specific proxies for mapping plausible regulatory activity across different cell types or conditions [1].</div>

				<div class="tab4collectionDescDetails">The Encyclopedia of DNA Elements (ENCODE) Consortium is an international effort aiming to build a comprehensive list of functional elements of the human genome. ENCODE generated maps of regulatory elements using DNA hypersensitivity assays, assays of DNA methylation, and immunoprecipitation of proteins that interact with DNA/RNA, followed by sequencing (i.e. modified histones, transcription factors, chromatin regulators, and RNA-binding proteins) [2].</div>

				<div class="tab4collectionDescDetails">[1] Cockerill, P. N. Structure and function of active chromatin and DNase I hypersensitive sites. FEBS J. 278, 2182â€“2210 (2011).</div>

				<div class="tab4collectionDescDetails">[2] ENCODE: Encyclopedia of DNA Elements (2015). https://www.encodeproject.org</div>

				<!-- <iframe src="https://hyperbrowser.uio.no/gsuite/datasets/3d820423f7c1d436/display/?preview=True" scrolling="no" width="900" height="2300"></iframe> -->
				</div>

				<div class = "welcomeAnalysisDetailsV2" > Analysis </div>



				<div class="tab4collectionDescDetails">This example examines the characteristics of ENCODE-generated DHS
					tracks accessible via HyperBrowser tool ''Create a remote GSuite from a public repository''. About
					450 tracks, encompassing over 150 different cell types and multiple methods of generating DHS
					profiles, are included in the initial track overview (each individual track corresponds to DNase I
					hypersensitive sites of a specific cell-type, treatment, method ... combination). In addition, a
					closer look is taken at a selected subset of cell-types and methods, exploring their effect on DHS
					profiling results.</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Analysis datasets - history </div>
				<div class="tab4collectionDescDetails_history">
					Annotated analysis history (including all the utilized data and operations) is available on a
					separate page:
					<a href="../u/hb-superuser/h/chromatin-accessibility-dnase-across-cell-types" target="_blank">
						Analysis history: Chromatin accessibility (DHS) across cell types</a>.
					(All results mentioned below refer to elements from this history page.)
				</div>




				<div class = "welcomeAnalysisDetailsV2V2" > - Overview of all tracks: </div>
				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetails">For the initial overview creation, all the tracks
							available via HyperBrowser tool "Create a remote GSuite from a public repository" and
							satisfying the following criteria were selected:</div>
						<div class="tab4collectionDescDetails">repository: ENCODE (UCSC and Ensembl)</div>
						<div class="tab4collectionDescDetails">Experiment (Assay) Type: DnaseSeq</div>
						<div class="tab4collectionDescDetails">File Type: narrowPeak</div>


						<div class="tab4collectionDescDetails_historyV3">
							<a a href="../u/hb-superuser/d/166ce9263f646cca/" target="_blank">Overview page for the 452-track GSuite for
								ENCODE DHS data </a> shows differences and similarities between tracks (e.g. with
							respect to genome/exone/repeat coverage or mutual overlap). It documents a considerable
							variability in peak count (77 458	- 405 844 peaks, median of 154 233.5 peaks) and average
							peak width (150 - 826 bp, median of 150 bp), which together determine the total base-pair
							coverage of the individual tracks (11 670 340	- 109 437 893 bp, median of 26 853 060 bp).
							To what degree should these differences be attributed to technical reasons?
						</div>
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Comparison of the effect of cell types and data processing
					methods on DHS profiles: </div>

					<div class="tab4collectionDescDetailsV2">

						<div class="tab4collectionDescDetails_historyV2">
							<div class="tab4collectionDescDetails">A subset of 3 cell-types (cell-lines representing
								different lineages), each with a triplet of alternative DHSs tracks (representatives of
								ENCODE's methods of DHSs determination), was created for a closer inspection.</div>

							<div class="tab4collectionDescDetails">The selected cell types and their
								characteristics:</div>


							<table class="tableEx">
							<tr><th>Cell</th><th> Lineage</th><th> Tissue</th><th> Karyotype</th></tr>

							<tr><td>H1-hESC</td><td> inner cell mass </td><td>embryonic stem cell</td><td> normal</td>
							</tr>

							<tr><td>GM12878 </td><td> mesoderm</td><td> blood </td><td>normal</td></tr>

							<tr><td>HeLa-S3 </td><td>ectoderm</td><td> cervix</td><td> cancer</td></tr>
							</table>

							<div class="tab4collectionDescDetails">
								(<a href="http://www.neuroepigenomics.org/encode/cellTypes.html" target="_blank">
								Read more about cell types used in the ENCODE project.</a>)</div>



								<div class="tab4collectionDescDetails">Multiple methods for determining DHS sites were
									used with each of the selected cell types:</div>

								<table class="tableEx">
								<tr><th>Method</th><th> Short description</th><th> Reference build/Analysis source</th>
									<th> Track name indicator</th></tr>

								<tr><td>a) "Uniform DNaseI HS"</td><td> DNaseI Hypersensitivity Uniform Peaks from
									ENCODE/Analysis, Integrated Regulation from UCSC </td><td>Human Genome Build 37
									(hg19): ENCODE Analysis Data at UCSC</td><td> "AwgDnaseUwduke"</td></tr>

								<tr><td>b) "Duke DNaseI HS" </td><td> Open Chromatin by DNaseI HS from
									ENCODE/OpenChrom(Duke University)</td><td> Human Genome Build 37 (hg19):
									ENCODE Production Data </td><td>"OpenChromDnase"</td></tr>

								<tr><td>c) "UW DNaseI HS" </td><td>DNaseI Hypersensitivity by Digital DNaseI from
									ENCODE/University of Washington</td><td> Human Genome Build 37 (hg19): ENCODE
									Production Data</td><td> "UwDnase"</td></tr>
								</table>

								<div class="tab4collectionDescDetails">
									(<a href="http://genome.ucsc.edu/ENCODE/downloads.html" target="_blank">Read more
									about data (types) produced by the ENCODE consortium.</a>)</div>

								<div class="tab4collectionDescDetailsDesc"> - 11-track subset overview</div>
							<div class="tab4collectionDescDetailsV3">
								<div class="tab4collectionDescDetails">11 of the original 452 tracks were included in
									a new GSuite collection based on their cell-type+method combination (9 tracks for
									all possible combinations plus two duplicate tracks; tracks from experiments
									involving additional cell culture treatments were not included).
									<a href="../u/hb-superuser/d/c86a1d755b1bb24d/" target="_blank">Overview statistics for the
										reduced GSuite</a> show that:</div>

								<div class="tab4collectionDescDetails">1. After ordering the tracks by base-pair
									coverage, it is apparent that the method of determining DHS sites (and not the
									cell type) is the primary cause of differences. Without any exception, tracks
									produced by method b) cover a higher proportion of the reference genome than tracks
									produced by method a), while the output tracks of method c) are grouped at the
									bottom of the ordering. The cell types are, however, consistently ranked within
									each group (even though the differences deciding about this ranking are sometimes
									small): H1-hESC covers more unique base-pairs than HeLa-S3, which in turn covers
									more unique base-pairs than GM12878. (statistic "Base-pair coverage")</div>

								<div class="tab4collectionDescDetails">2. The average peak width is strictly
									method-dependent, with method b) being markedly different compared to methods
									a) and c), which are practically identical regarding this measure. (statistic
									"Segment length average")</div>

								<div class="tab4collectionDescDetails">3. Method a) generates more peaks than the other
									methods. The GM12878 cell-line produces fewer peaks when compared to other cell
									types (when processed by the same method). (statistic "Segment count")</div>
							</div>
								<div class="tab4collectionDescDetailsDesc"> - Clustering of tracks</div>
							<div class="tab4collectionDescDetailsV3">
								<div class="tab4collectionDescDetails">How would the 11 tracks cluster considering the
									apparent dependency of DHS profiles on the method of their creation?</div>

								<div class="tab4collectionDescDetails">When clustered with a method based on
									"similarity of positional distribution along the genome" (using the default binning
									settings),
									<a href="../u/hb-superuser/d/56d124232a71c09f/cluster_tracks_result_figure.pdf"
									   target="_blank"> the tracks cluster primarily by analysis method</a>.
									When clustered with a method based on "direct sequence similarity" (using pairwise
									overlap enrichment),
									<a href="../u/hb-superuser/d/066d0b070a1991a8/cluster_tracks_result_figure.pdf"
									   target="_blank">the tracks cluster primarily by cell type</a>.
									These results suggest that while the examined DHS determination methods suffer from
									biases with respect to peak density, true biological signal comes
									through on base-pair level.</div>
							</div>
								<div class="tab4collectionDescDetailsDesc"> - Conclusions</div>
							<div class="tab4collectionDescDetailsV3">
								<div class="tab4collectionDescDetails">When working with ENCODE, attention to tracks'
									origin should be paid and special care should be taken when drawing conclusions
									from comparisons of tracks which do not come from the same production source.
									Initial exploration of tracks' properties is highly recommended.</div>

								<div class="tab4collectionDescDetails">Choice of track comparison methods can have a
									prominent influence on the results - an appropriate analysis approach needs to be
									selected on case-by-case basis.</div>



							</div>

						</div>
					</div>
			</div>




			<div class="tab4collectionTitle" id="T7">
				<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" height="20" width="20" id="B7" class='-off' /></div>
						- Disease-associated SNPs in miRNAs
			</div>
			<div class="tab4collectionDesc" id="D7">
				<div class="show_hide" id="T71">
					<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" id="B71" height="20" width="20"  class='-off' /></div>
					- Background
				</div>
				<div class="sliding" id="D71">
					<div class="tab4collectionDescDetails">MicroRNAs (miRNAs) are key regulators of biological processes and are heavily
						involved in human diseases such as cancers (Alvarez-Garcia and Miska 2005).  The deregulation of miRNA genes
						is thought to be the major source of this pathogenic development but mechanisms of how this deregulation is
						facilitated in the cells, remains largely unknown.
					</div>
					<div class="tab4collectionDescDetails">Looking at overlapping regions of miRNAs and disease related SNPs using
						GSuite tools could help discover mutations in miRNA genes that have effects on their target-range, or
						mutations in miRNA regulatory motifs that can have effects on processing efficiency (Auyeung, et al. 2013)
						and might be connected to the pathology of the disease.
					</div>
					<div class="tab4collectionDescDetails">In order to achieve an understanding of how variable miRNA parts are in
						general, we did the same analysis with all common SNPs.
					</div>
					<div class="tab4collectionDescDetails">[1] Alvarez-Garcia, I., and E. A. Miska, MicroRNA functions in animal
						development and human disease. Development 132(21):4653-62, 2005.
					</div>
					<div class="tab4collectionDescDetails">[2] Auyeung, V. C., et al., Beyond secondary structure: primary-sequence
						determinants license pri-miRNA hairpins for processing. Cell 152(4):844-58, 2013.
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2" > Datasets used in the analysis </div>
				<div class = "welcomeAnalysisDetailsV2V2" > - Common SNPs </div>
				<div class="tab4collectionDescDetails_history">
					Common SNPs refers to variants that appear to be reasonably common in the general population, and are less likely associated with severe genetic diseases due to effects of natural selection. The dataset of common SNPs (SNP142common) used in this analysis is downloaded from UCSC, and contains information of a subset of single nucleotide polymorphisms that have a minor allele frequency of at least 1 % and are mapped to a single base pair in the reference genome assembly (http://genome.ucsc.edu/). Small insertions and deletions (indels) are also included in the original dataset, but are filtered out in our analysis.
				</div>
				<div class = "welcomeAnalysisDetailsV2V2" > - Disease related SNPs </div>
				<div class="tab4collectionDescDetails_history">A suite of tracks of Tag SNPs associated with 1362 different traits
					were downloaded from the GWAS (Genome-Wide
					Association Studies) catalogue. As SNPs are not independent, but tend to be coinherited in groups,
					only a representative proportion of the disease associated SNPs (Tag/lead/index SNPs) are covered in
					this data set. To include all the SNPs located nearby the Tag SNPs, we extended each GWAS track of disease
					associated SNPs to incorporate all common SNPs (SNP142common, described above) within 25kb from annotated Tag
					SNPs. In the final analysis result we excluded trait tracks having no overlap with miRNAs, ending up with a
					subset of 78 traits reported in the result tables.
				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - miRNA data </div>
				<div class="tab4collectionDescDetails_history">
					<div class="tab4collectionDescDetails">By manually curating all 1'881 putative miRNA entries currently
						deposited for human in the online repository miRBase (v21) [1], a subset of 521
						manually curated miRNA genes that fulfill structural criteria for the annotation of
						miRNA genes was derived (MirGeneDB.org) [2].

						Here, we use this refined human miRNA complement that is annotated in all parts
						(primary miRNA, precursor-miRNA, mature miRNA including seed, loop sequence, star sequence including
						seed and its flanking regions (30nts) on both ends), with tracks for each miRNA part constituting
						a miRNA suite.
					</div>

					<img src="../static/hyperbrowser/images/div/mirna_illustration.png" title="Figure 1: Illustration of the different parts of a miRNA" style="width: auto; height: 400px;"></img>
					<div class="tab4collectionDescDetails">Figure 1: Illustration of the different parts of a miRNA</div>

					<div class="tab4collectionDescDetails">[1] Kozomara, A., and S. Griffiths-Jones, miRBase: annotating high
						confidence microRNAs using deep sequencing data. Nucleic Acids Res 42(1):D68-73, 2014.
					</div>
					<div class="tab4collectionDescDetails">[2] Fromm, B., et al., A Uniform System for the Annotation of
						Vertebrate microRNA Genes and the Evolution of the Human microRNAome. Annu Rev Genet 49:213-42., 2015.
					</div>

				</div>

				<div class = "welcomeAnalysisDetailsV2" > Analysis using GSuite Tools </div>
				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetails">Using GSuite Tools we examined whether the genomic regions of the
							different microRNA parts, as well as whole miRNAs, overlap with commonly occurring SNPs (from UCSC)
							and SNPs that are associated with different traits (from GWAS catalogue). We compiled the data sets
							described above and analyzed:
						</div>
						<div class="tab4collectionDescDetails">1. The intersection between each GWAS extended SNP dataset (GSuite)
							against each data set of a particular miRNA part or whole miRNA (GSuite). Annotated analysis history is
							available on a separate page:
							<a href="../u/hb-superuser/h/disease-associated-snps-in-mirnas" target="_blank">Disease-associated SNPs in miRNAs</a>
						</div>

						<div class="tab4collectionDescDetails">2. The intersection between the common SNP dataset (track) against each
							dataset of a particular miRNA part or whole miRNA (GSuite). Annotated analysis history is available on a
							separate page:
							<a href="../u/hb-superuser/h/mirna-vs-expanded-snps-v3" target="_blank">Common SNPs in miRNAs</a>
						</div>
					</div>
				</div>
				<div class = "welcomeAnalysisDetailsV2V2" > - Results: </div>
				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetailsDesc"> - Common SNPs</div>
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">Of the 12'969'631 common SNPs for human, 217 SNPs were
								observed in 166 of the 521 miRNAs. This corresponds to a SNP density of 2,6 SNPs/10 KB
								within pri-miRNA. When compared to other parts of the human genome such as the full genome
								(4,6 SNPs/10 KB), introns (4,4 SNPs / 10 KB) and exons (3,9 SNPs / 10KB) this value is
								significantly lower underlining the strong selective pressure against changes of miRNAs in
								general. More specifically we observe a gradual depletion of SNPs for parts of the miRNA
								precursor depending on their importance for miRNA function (figure 1).
							</div>

							<img src="../static/hyperbrowser/images/div/mirna/Figure_1.png" style="width: 100%; height: auto;"></img>
							<div class="tab4collectionDescDetails">Figure 1: distribution of common SNPs across different
								parts of the human genome. MiRNA loci show significantly lower SNPs/10 Kbps than the rest
								of the genome.
							</div>
						</div>

						<div class="tab4collectionDescDetailsDesc"> - Disease related SNPs</div>
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">
								<div class="tab4collectionDescDetails_historyV3">
									<a href="../u/hb-superuser/d/3660b276e57b0ab8" target="_blank">Of
										the 3'266'867 disease related (unique) SNPs in 78 GWAS datasets, 62 were
										observed in 50 of the 521 miRNAs.</a>
								</div>
								Of these 50 miRNAs, 10 were co-matures, 19 5' matures and 21 were 3' matures respectively.
							</div>
							<div class="tab4collectionDescDetails">The SNPs were distributed in all parts of miRNAs
								but showed the highest prevalence in the 5' and 3' flanking regions of the miRNAs
								(30 and 19 reported SNPs respectively); possibly within regulatory motifs (UG; CNNC).
								In line with the observation from the common SNPs the star-sequences showed the
								next-highest prevalence (5) followed by identical numbers (4) for loop and matures.
							</div>

							<img src="../static/hyperbrowser/images/div/mirna/Figure_2.png" style="width: 100%; height: auto;"></img>
							<div class="tab4collectionDescDetails"> Figure 2: distribution of common and disease-related
								SNPs across different parts of the human genome.
							</div>
						</div>

						<div class="tab4collectionDescDetailsDesc"> - Relative SNP density</div>
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">That the total number of disease related SNPs is
								significantly smaller than common SNPs is known. It was however interesting to see
								that the abundance of SNPs between the different parts of the genome seemed so similar.
								A more detailed comparison of the relative density of SNPs (normalized for the ratio of
								total SNPs to the respective genomic parts in the common SNP dataset) in the
								disease-related SNPs shows an enrichment of SNPs in exons and 5' flanking regions
								of miRNAs.
							</div>
							<img src="../static/hyperbrowser/images/div/mirna/Figure_3.png" style="width: 100%; height: auto;"></img>
							<div class="tab4collectionDescDetails">Figure 3: Normalized distribution of common and
								disease-related SNPs across different parts of the human genome show deviations.
							</div>
						</div>
					</div>
				</div>
			</div>

			<div class="tab4collectionTitle" id="T8">
				<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" height="20" width="20" id="B8" class='-off' /></div>
						- Chromatin accessibility and mutational profiles of tumors
			</div>

			<div class="tab4collectionDesc" id="D8">

				<div class="show_hide" id="T81">
					<div class="imgBackground">
						<img src="hyperbrowser/images/div/DNAlying.png" id="B81" height="20" width="20"  class='-off' />
					</div>
					- Background
				</div>

				<div class="sliding" id="D81">
					<div class="tab4collectionDescDetails">Tumor genomes are highly heterogeneous with respect to
						their mutational burden. This heterogeneity can arise from multiple sources. A common
						source is exposure to mutagens, such as ultraviolet radiation in malignant melanoma and
						tobacco in lung cancer [1]. In addition, it has recently been shown that the epigenomic
						profile (chromatin accessibility, histone modifications etc.) of a tumor's cell-of-origin is
						highly associated with its somatic mutational landscape [2]. Importantly, such findings may
						ultimately allow for accurate prediction of cell-of-origin of a cancer based on the somatic
						mutation profile. In this example usage of GSuite, we will investigate the strong statistical
						association that was found for chromatin accessibility in melanocytes and the somatic mutation
						density in malignant melanoma [2]
						(see <a href="http://www.nature.com/nature/journal/v518/n7539/fig_tab/nature14221_F1.html" target="_blank">
							wym-1462355009093 </a>). We will demonstrate through a fully reproducible
						workflow how this question may be approached within the GSuite framework.
					</div>
					<div class="tab4collectionDescDetails">[1] Lawrence M. et al. Mutational heterogeneity in cancer and the
						search for new cancer-associated genes. Nature 499, 214-218 (2013)
					</div>
					<div class="tab4collectionDescDetails">[2] Polak, P. et al. Cell-of-origin chromatin organization shapes
						the mutational landscape of cancer. Nature 518, 360-364 (2015).
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2" > - Datasets used in the analysis: </div>

				<div class="tab4collectionDescDetails_history">
					<div class="tab4collectionDescDetails">In order to address the question we utilised the following datasets: </div>
					<div class="tab4collectionDescDetails">- Whole-genome somatic mutational profiles in melanoma (simple somatic
						mutations, ICGC)
					</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">* SKCA-BR (Skin adenocarcinoma, Brasil) </div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">* MELA-AU (Skin cancer, Australia) </div>
					<div class="tab4collectionDescDetails">- Chromatin accessibility tracks across cell types (DnaseSeq from ENCODE) </div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">* A549, HepG2, MCF-7, Adult_CD4+, Caco-2,
						COLO829, Cerebellum, Fibrobl, H1-hESC, HepG2, Hepatocytes, LNCaP, Medullo, Melano (epidermal
						melanocytes), Mel-2183 (melanoma cell line), Naive_Bcell, Osteobl, PANC-1, PanIslets, SKMC, T47-D
					</div>
					<div class="tab4collectionDescDetails">In order to investigate the reported relationship in an unbiased
						manner, we included a rich set of cell types/cell lines for the chromatin accessibility data.
					</div>

				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Analysis/comments: </div>

				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">The <a href="../u/hb-superuser/h/gsbe---chromatin-accessibility-and-mutational-profiles-of-tumors" target="_blank"> analysis </a>
								was roughly divided in three parts:</div>
							<div class="tab4collectionDescDetails">1) Creation of two mutation datasets, each containing
								somatic variants in skin cancer donors (from the International Cancer Genome Consortium, ICGC)
							</div>
							<div class="tab4collectionDescDetails">2) Creation of a GSuite of chromatin accessibility datasets
								for a diversity of cell lines/ cell types(from ENCODE)
							</div>
							<div class="tab4collectionDescDetails">3) Determine the GSuite tracks in 2) that most strongly
								coincide with the two mutation datasets in 1), i.e. which cell-type specific chromatin
								accessibility profile is most strongly correlated to the mutation profile of skin cancer?
							</div>
							<div class="tab4collectionDescDetails">Steps 2) and 3) were readily accessible within the main
								functionality of the GSuite framework. The establishment of somatic datasets from ICGC (step 1))
								required some additional Galaxy steps (manipulation/filtering) because the raw data contained
								duplicate entries (due to variant overlap in different transcript isoforms).

								With respect to the key step in our workflow (3)), we used the analysis available through
								Statistical analysis of GSuites - Determine GSuite tracks coinciding with a target track.
								As the track-to-track similarity measure, we chose the Correlated bin coverage, using 1Mb as
								our bin size (this to resemble the analysis [1]). Here, the chromatin datasets acted as a
								GSuite, while the mutation datasets were each represented as GTracks. Using this approach, we
								obtained a similarity score (Correlated bin coverage?) for each of the cell-specific chromatin
								accessibility tracks in relation to the mutation datasets. Overall, we found that for both the
								SKCA-BR and MELA-AU datasets, chromatin accessibility tracks for epidermal melanocytes and the
								Mel-2183 melanoma cell line were ranked as the most dissimilar ones. This finding supports the
								strong negative correlation that was reported by Polak et al. In conclusion, this example of the
								GSuite framework demonstrates how a track-based similarity of chromatin accessibility tracks and
								cancer mutation datasets can highlight the most likely cell-of-origin of a cancer genome.
							</div>
							<div class="tab4collectionDescDetails">[1] Polak, P. et al. Cell-of-origin chromatin organization
								shapes the mutational landscape of cancer. Nature 518, 360-364 (2015).
							</div>
						</div>
					</div>
				</div>
			</div>

			<div class="tab4collectionTitle" id="T2">
				<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" height="20" width="20" id="B2" class='-off' /></div>
						- Investigating the co-localisation between  transcription factors and histone marks
			</div>

			<div class="tab4collectionDesc" id="D2">

				<div class="show_hide" id="T21">
					<div class="imgBackground">
						<img src="hyperbrowser/images/div/DNAlying.png" id="B21" height="20" width="20"  class='-off' />
					</div>
					- Background
				</div>

				<div class="sliding" id="D21">
					<div class="tab4collectionDescDetails">Expression of genes is a highly complex process where transcription
						factors and epigenetic modifications play important roles. Transcription factors are able to alter
						the epigenetic environment of target genes, however, their binding is also influenced by already
						existing marks. Hence, by investigating which epigenetic marks transcription factors co-localise with
						it is possible to obtain preliminary information about their regulatory properties.
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2" > - Preface: </div>
				<div class="tab4collectionDescDetails_history">
					<div class="tab4collectionDescDetails">In this example we are going to investigate the genomic co-localisation
						between a subset of transcription factors and epigenetic marks known as histone modifications
						from the K562 leukemia cell line generated by the ENCODE consortium [1]. The epigenetic mark
						tri-methylation of histone 3 lysine 4 (H3K4me3) and acetylation of the same histone at lysine
						9 and 27 (H3K9ac and H3K27ac) are found to correlate with active transcription while the tri-methylation
						of lysine 9 and 27 (H3K9me3 and H3K27me3) with repression [2-4]. The workflow outlined can be used
						to investigate the relationship between additional genomic interacting proteins and features,
						such as DNA methylation and genomic mutations. The data used in the example are genomic areas
						enriched for the transcription factor or histone modification in question, known as peaks, generated
						by the technique Chromatin Immunoprecipitation coupled with next generation sequencing (ChIP-Seq).
					</div>
					<div class="tab4collectionDescDetails">[1] ENCODE Project Consortium, Bernstein BE, Birney E, Dunham I,
						Green ED, Gunter C, et al. (2012) An integrated encyclopedia of DNA elements in the human genome.
						Nature 489: 57â€“74
					</div>
					<div class="tab4collectionDescDetails">[2] Barski A, Cuddapah S, Cui K, Roh T-Y, Schones DE, Wang Z,
						et al. (2007) High-resolution profiling of histone methylations in the human genome. Cell
						129: 823â€“837.
					</div>
					<div class="tab4collectionDescDetails">[3] Mikkelsen TS, Xu Z, Zhang X, Wang L, Gimble JM, Lander ES,
						et al. (2010) Comparative epigenomic analysis of murine and human adipogenesis.
						Cell 143: 156â€“169.
					</div>
					<div class="tab4collectionDescDetails">[4] Barth T and Imhof A (2010) Fast signals and slow marks:
						the dynamics of histone modifications., 35(11), 618â€“626.
					</div>

				</div>

				<div class = "welcomeAnalysisDetailsV2" > - Datasets used in the analysis: </div>
				<div class="tab4collectionDescDetails_history">
					<div class="tab4collectionDescDetails">Annotated analysis history (including all the utilized data and
						operations) is available on a <a href="../u/hb-superuser/h/imported-imported-transcription-factors-mads2" target="_blank"> separate page </a>. (All results mentioned below refer
						to elements from this history page.)
					</div>
					<div class="tab4collectionDescDetails">Initial two GSuites for the subset of
						histone modifications and transcription factors, respectively, were created using
						"Create a GSuite from an integrated catalog of genomic datasets" satisfying
						the following criteria were selected:
					</div>

					<div class="tab4collectionDescDetails"><b>Histones:</b></div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Track category: Histone variants and
						modifications (search by cell/tissue type)
					</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Sub-category: K562 Leukemia cell line</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Database:  All original tracks</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">File type: narrowPeak</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Datatype: Others</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Selected subset of tracks manually</div>

					<div class="tab4collectionDescDetails"><b>Transcription factors:</b></div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Track category: Transcription factor
						binding sites (search by cell/tissue type)
					</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Sub-category: K562 Leukemia cell line</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Database:  All original tracks</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">File type: narrowPeak</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Data type: narrowPeak</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;">Selected subset of tracks manually</div>

					<div class="tab4collectionDescDetails">For downstream analysis the downloaded files were used.</div>
					<div class="tab4collectionDescDetails">In order to count the number of times each transcription factor
						overlap with a histone modifications the GSuite containg the peaks for the transcription factors
						was modified to contain only the middle point of each peak using "Modify primary tracks referred
						to in a GSuite":
					</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Operation:</b> Expand all points and segments equally </div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Parameter:</b> typed: "middle"</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Change file suffix:</b> "Yes" and changed suffix to "bed"</div>

					<div class="tab4collectionDescDetails">To investigate how the transcription factors colocalizes with the
						different histone modifications, the two GSuites were preprocessed for analysis using
						"Preprocess a GSuite for analysis" and secondly a co-localization analysis were setup using
						"Determine coinciding track combinations from two suites" selecting the following parameters:
					</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Query track:</b> GSuite with transcription factors containing the middle point of each peak.</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Reference track:</b> GSuite with Histone modifications</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Genome:</b> hg19</div>
					<div class="tab4collectionDescDetails" style="margin-left: 20px;"><b>Similarity measure:</b> Forbes coefficient: ratio of observed to expected overlap.</div>

				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Conclusion: </div>

				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">The colocalization shows that the transcription factors
								overlaps with various degree with the histone marks associated with active transcription
								(H3K4me3, H3K9ac and H3K27ac) while only a small fraction is overlapping with the histone
								marks linked to repression (H3K9me3 and H3K27me3).
							</div>
						</div>
					</div>
				</div>
			</div>



			<div class="tab4collectionTitle" id="T4">
				<div class="imgBackground"><img src="hyperbrowser/images/div/DNAlying.png" height="20" width="20" id="B4" class='-off' /></div>
						- Exploring transcription factor co-occurrence using two alternative measures of similarity
			</div>

			<div class="tab4collectionDesc" id="D4">

				<div class="show_hide" id="T41">
					<div class="imgBackground">
						<img src="hyperbrowser/images/div/DNAlying.png" id="B41" height="20" width="20"  class='-off' />
					</div>
					- Background
				</div>

				<div class="sliding" id="D41">
					<div class="tab4collectionDescDetails">GATA1 is a zinc-finger type transcription factor (TF) with a key
						role in regulating gene programs during hematopoiesis, where it induces megakaryocytic and
						erythroid commitment and simultaneously prevents granulocyte-monocyte and lymphoid development
						(Kitajima et al. PMID: 16543218, Welch et al. PMID: 15297311; Ferreira et al. PMID: 15684376).
						GATA1 is sequence-specific DNA-binding protein recognizing the GATA box motif (WGATAR) found in
						regulatory sites controlling its target genes (Trainor 1996, PMID: 8628290).
					</div>
					<div class="tab4collectionDescDetails">Approximately 20,000 sites with GATA1 bound are found in the genome of
						erythroid cells (Ulirsch JC, et al., PMID: 25521328). A recent study which used high-resolution
						ChIP-exo to map GATA1 and TAL1 across the mouse genome, identified âˆ¼10,000 GATA1 and âˆ¼15,000 TAL1,
						of which âˆ¼4,000 locations were bound by both GATA1 and TAL1 (PMID: 26503782). The current understanding
						is that key complexes are formed by combinatorial occupancy patterns of erythroid TFs leading to the
						induction of key erythroid genes.
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2" > - Datasets used in the analysis: </div>
				<div class="tab4collectionDescDetails_history">
					<div class="tab4collectionDescDetails">
						To perform a similarity measure of GATA1 in K562 cells, we collected 318 ChIP-seq datasets
						from ENCODE (PMID 22955616) and downloaded to history. We then performed similarity testing
						of GATA1 ChIP-seq track (Snyder lab, Stanford) against the 317 experimental datasets
						using the Forbes and Jaccard similarity measures.
					</div>
				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Analysis datasets - history </div>
				<div class="tab4collectionDescDetails_history">
					Annotated analysis histories (including all the utilized data and operations) are available on
					separate pages for the <a href="../u/hb-superuser/h/gata1-example---the-effect-of-similarity-measure" target="_blank"> analysis leading to ranking of tracks </a>
					and the <a href="../u/hb-superuser/h/gata1-example---subsampling-tracks" target="_blank">quantitative analysis of
					correlation between similarity measures and track sizes </a>. For both histories, the history elements
					are annotated with details of the performed analysis and their main interpretation.

				</div>

				<div class = "welcomeAnalysisDetailsV2V2" > - Analysis and conclusions: </div>
				<div class="tab4collectionDescDetailsV2">
					<div class="tab4collectionDescDetails_historyV2">
						<div class="tab4collectionDescDetailsV3">
							<div class="tab4collectionDescDetails">We generated two separate rankings of the 317 chip-seq tracks,
								based on their similarity against the GATA1 track according to the Forbes and Jaccard similarity
								measures, respectively. In addition to exploring the biological interpretations of the two different
								rankings, we performed a quantitative analysis of how similarity values correlated with the size of
								tracks and how the (Forbes versus Jaccard) similarity values changed through subsampling of elements
								from the tracks.
							</div>
							<div class="tab4collectionDescDetails">The two alternative measures (Forbes and Jaccard) resulted in
								very different rankings of similarity to GATA1. With the Forbes measure, another replicate of
								GATA1 was ranked first, along with several other potentially interesting TFs at the top, but
								without expected co-binding TFs like TAL1 being among the most highly ranked TF. The ranking
								according to the Jaccard measure were clearly influenced by size of tracks, with other
								replicates for GATA1 not being among the highest ranked TF. At the same time, the expected
								co-binding TF TAL1 was ranked among the top tracks, along with several tracks for GATA2.
							</div>
							<div class="tab4collectionDescDetails">To further explore the influence of track size on the
								measures, we performed a quantitative analysis of the Forbes and Jaccard similarity values.
								We selected 291 ChIP-seq tracks that had at least 1000 peaks, and made separate Scatterplots
								of Forbes similarity versus track size and Jaccard similarity versus track size.
								The Jaccard similarity values was seen to be strongly correlated with track size.
								To further determine whether this was directly connected to track size, we randomly
								subsampled a set of 1000 peaks for each track. The Forbes values were essentially unchanged
								by this subsampling, while the Jaccard values changed substantially. This shows that the
								Jaccard measure is very sensitive to the number of elements in tracks, and that care must be
								taken to ensure that biological interpretations are not dominated by technicalities related to
								the number of discovered peaks.
							</div>
						</div>
					</div>
				</div>


			</div>



		</div>
	</div>

	<div id="tab5" class="tab">

	    <div class="document">
			<div class="main">

				<div class="contact">

				    <b>Note:</b>
				    <p>
				    This is a specialized version of The Genomic Hyperbrowser,
				    focusing on new functionality for analyzing collections of genomic track.
				    More information on the general HyperBrowser functionality can be
				    found on the <a href="https://hyperbrowser.uio.no/hb/" target="_blank">main project site</a>.
				    </p>
				    <b>Contact information</b>
				    <p>
				    If you have any questions, requests or comments, please use our discussion forum (available from the <a minsizehint="-1" href="https://sites.google.com/site/hyperbrowserhelp/?tool_id=hb_help">Help</a> menu) or send us an email to <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@stseuqer-resworbrepyh</a>&#x200E;. We are happy to respond and welcome collaborative projects that help further develop our system. If you encounter any bugs, please let us know via an email to
				  <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@sgub-resworbrepyh</a>.</p>
				    <p>If you register as a user, you will be able to subscribe to the <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">on.oiu.tisu@ofni-resworbrepyh</a> mailing list, which serves for distribution of system-related announcements and administrative messages. A backlog of announcements is available from the Help menu. The Help menu also contains citation information.</p>

				</div>
			</div>
		</div>
		<p class="blocktext">
		This project is being developed by the Norwegian bioinformatics community as an open-source project under the GPL  license v3,
		supported by various national and local bodies.
		</p>
		<p class="blocktext">
		  For further information on Norwegian bioinformatics, please visit <a href="http://www.bioinfo.no">www.bioinfo.no</a>.
		</p>

		<div class="section">
		  <img src="hyperbrowser/images/logo/HB_support.png" class="center">
		</div>

	</div>

    </div>
</div>





</body>
</html>
