<!DOCTYPE html><html><head><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"><script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script><link rel="stylesheet" href="welcome_project.css"><meta charset='UTF-8'><title>coloc-stats: a unified web interface to perform co-localization analysis of genomic features</title></head><body><div class='content'>
    <div class ="tabs">
        <ul class ="nav nav-tabs" id="tab-links">
            <li class="active" value="tab0">
                <a data-toggle="tab" href="#menu0">coloc-stats</a>
            </li>
            <li  value="tab1">
                <a data-toggle="tab" href="#menu1">Documentation</a>
            </li>
            <li  value="tab2">
                <a data-toggle="tab" href="#menu2">FAQ</a>
            </li>
            <li  value="tab3">
                <a data-toggle="tab" href="#menu3">About</a>
            </li>

        </ul>
    </div>
    <div class="tab-content">
        <div id="menu0" class="tab-pane fade  in active">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <h1 class="projectTitle">coloc-stats</h1>
                    <div class = 'projectDescSingle'>
                        a unified web interface to perform <a href="#co-loc-analysis"><font color="#0645AD">co-localization analysis</font></a> of genomic features
                    </div>
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                            <h4>Get started by:</h4>
                            <ol>
                                <li>
                                    <a  class = "button" href="javascript:void(0)" onclick="event.preventDefault(); jQuery('#tool-panel-upload-button', window.parent.jQuery(window.parent.document)).trigger('click');">Uploading datasets</a> of genomic locations (bed-files)
                                </li>
                                <li>
                                    Go to the <a class="button" id="coloc-tool" href="../tool_runner/data_source_redirect?tool_id=hb_conglo_proto_tool">coloc-stats tool</a> to analyze your datasets by multiple methods
                                </li>
                                <li>
                                    View your results (by clicking the eye icon on the history element to the right)
                                </li>
                            </ol>

                        </div>
                        <div class ="projectDesc" style="margin: 20px 0px; font-style: italic;">
                            If you are new to Galaxy, you can follow a <a class="button" href="https://galaxyproject.org/tutorials/g101/" target="_top">brief tutorial</a>
                        </div>
                        <div class ="projectDesc">
                            <b>You can also:</b>
                            <ul>
                                <li>
                                    <a class="button" href="https://hyperbrowser.uio.no/coloc-stats/u/gsandve/h/coloc-stats-example-analysis" target="_top">View an example analysis</a>
                                </li>
                                <li>
                                    <a class="button" href="https://hyperbrowser.uio.no/coloc-stats/u/gsandve/h/coloc-stats-sample-data" target='_top'>Load a history</a> with small sample datasets (in the next page, click "import history")
                                </li>
                                <li>
                                    Look at the <a href='https://vimeo.com/249248452' target='_top'><font color="#0645AD">screencasts</font></a> below on how to use the system
                                </li>
                            </ul>
                        </div>
                        <div class ="projectDesc" style="margin: 20px 0px; font-style: italic;">
                            Please feel free to <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">contact us</font></a> with any questions.
                        </div>
                        <!--<div class ="projectDesc">-->
                            <!--<p class="paragraph">-->
                                <!--Several generic and specialized tools are available to perform co-localization analysis, they vary in the statistical methodology and analysis approaches, thus potentially providing different conclusions for the same research question. As the findings of co-localization analysis are often the basis for further follow-up experimental investigations, assessing the robustness of the findings is warranted. However, there exists no unified interface to perform co-localization analysis across a multitude of analytical methods and method-specific options (e.g., test statistics, resolution, null models, and whether or not they can handle local genomic heterogeneities and co-localizing confounding features). To address this gap, we propose a coloc-stats, single interface for multiple previously published co-localization analysis methods. </p><p class="paragraph">coloc-stats allows a user to find a method that supports their experimental requirements, and when multiple methods meet the requirements, assess the robustness of conclusions by running and inspecting results across methods. Additionally, we extend methods that only support a pair of tracks to consider track collections.-->
                        <!--</p>-->
                        <!--</div>-->
                    </div>
                </div>
            </div>
        </div>
        <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'example'><h4 class = 'exampleTitle'>Screencast:quick tutorial</h4>
                    <!–– <video controls="controls" width="800" height="600"
                    name="Video Name" src="images/gettingStartedWithColocStats.mov">
                    <!–– </video>
                    <iframe src="https://player.vimeo.com/video/249248452" width="800" height="600" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
                    </div>
                </div>
                </div>
            </div>
        </div>
                        
        <!--<div class="footer">-->
            <!--<div class = 'contactInfo'>-->
                <!--<div class = 'contactInfoDesc'>-->
                    <!--All questions, requests or comments can be redirected to <a href='mailto:geirska@ifi.uio.no' target='_top'>Geir Kjetil Sandve.</a>-->
                <!--</div>-->
            <!--</div>-->
        <!--</div>-->
        </div>



        <div id="menu1" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDescSingle'>
                            <h3>Documentation</h3>

                            <div class = 'projectDesc'>
                                ........
                            </div>
                        </div>
                        <div class = 'projectDescSingle' >
                            <h3>General</h3>

                            <div class = 'projectDesc'>
                                <h4 id="co-loc-analysis">Co-localization analysis</h4>
                                <p>High-throughput sequencing methods assay various features of the genome
                                        including regulatory elements, transcription factor binding sites (TFBS),
                                        transcripts and so on. Functionally related genomic features often
                                        co-occur within a genomic sequence (for e.g., co-occurrence of TFBS).
                                        One important way of determining whether genomic features are functionally
                                        related, is to search for their significant co-localization
                                        (based on overlap or proximity). The methodology that determines a
                                        significant co-localization of genomic features is often referred to as
                                        co-localization analysis or alternatively as co-occurrence analysis or
                                    region set enrichment analysis.</p>
                                <p></p>
                                <h4>Basic overview of a typical co-localization analysis</h4>
                                <p>Genomic features are mapped on the reference genome using chromosomal
                                        coordinates (e.g., chr1:100-1000), which are simply the start position
                                        and end position of a sequence of nucleotides on a chromosome. Such a
                                        representation of nucleotide sequence is referred to as genomic regions
                                        or genomic intervals. One of the main outputs of genomic assays is a
                                        list of genomic regions and a collection of sets of genomic regions are
                                        referred to as genomic tracks or region sets. Arithmetic set operations
                                        are performed between genomic tracks to determine the amount of overlap
                                        (overlapping sequence nucleotides) or spatial proximity (e.g. geometric
                                        distance between two genomic regions). This is followed by statistical
                                        testing to determine whether the observed overlap or spatial proximity
                                    is likely due to chance.</p>
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h3>Input</h3>

                            <div class = 'projectDesc'>
                                ........
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h3>File formats</h3>

                            <div class = 'projectDesc'>
                                Coloc-stats supports files only in BED format. BED is a standard file format to store genomic regions, where the first three fields are required to be chr, start and end. Please refer to the BED file format described on the UCSC browser page (https://genome.ucsc.edu/FAQ/FAQformat.html#format1)
                            </div>
                        </div>
                        <div class = 'projectDescSingle'>
                            <h3>Invalid file formats</h3>

                            <div class = 'projectDesc'>
                                Coloc-stats does not accept any other genomic track file formats other than BED format for now. Also, the BED format is required to adhere strictly to the file format specifications.
                            </div>
                        </div>

                        <div class = 'projectDescSingle'>
                            <h3>File size limitations</h3>
                            <div class = 'projectDesc'>
                                We are not aware of any file size limitations as of now.
                            </div>
                        </div>

                        <div class = 'projectDescSingle'>
                            <h3>Genome assemblies</h3>

                            <div class = 'projectDesc'>
                                Coloc-stats does not accept any other genomic track file formats other than BED format for now. Also, the BED format is required to adhere strictly to the file format specifications.
                            </div>
                        </div>


                    </div>
                </div>
            </div>
        </div>
    </div>

        <div id="menu2" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll" style="padding-left: 0px">
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                             <dl id="faqs">
                                 <h3>Frequently asked questions</h3>
                                 (click on the question to see the answer)
                                 <p></p>
                                 <h4>## Why coloc-stats? When coloc-stats? How coloc-stats?</h4>
                                 <p></p>
                                 <ol>
                                     <li>
                                <dt><b>I know some co-localization analysis tools. What is the purpose of a new tool?</b></dt>
                                <dd>
                                    The existing co-localization analysis tools vary in the statistical methodology and analysis approaches,
                                        thus potentially providing different conclusions for the same research question. As the findings of
                                        co-localization analysis are often the basis for further follow-up experimental investigations, assessing
                                        the robustness of the findings is warranted. However, there exists no unified interface to perform co-localization
                                        analysis across a multitude of analytical methods and method-specific options (e.g., test statistics, resolution,
                                        null models, and whether or not they can handle local genomic heterogeneities and co-localizing confounding features).
                                        To address this gap, we propose coloc-stats, a single interface for multiple previously published co-localization analysis methods.
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>What can I achieve more with coloc-stats, than just running other co-localization analysis tools?</b></dt>
                                <dd>
                                    <div class = 'projectDesc'>
                                        Coloc-stats allows a user:
                                    </div>
                                    <ul>
                                        <li>
                                            To make the most appropriate analysis choices given their research question.
                                        </li>
                                        <li>
                                            Find a relevant method (amongst several) that supports their analysis choices.
                                        </li>
                                        <li>
                                            To assess the robustness of conclusions by running and inspecting results across several methods.
                                        </li>
                                        <li>
                                             Extend methods that only support a pair of tracks to consider track collections.
                                        </li>
                                    </ul>
                                </dd>
                                    </li>
                                     <p></p>
                                     <li>
                                <dt><b>When and why should I use coloc-stats?</b></dt>
                                <dd>
                                    <div class = 'projectDesc'>
                                        Coloc-stats is useful when:
                                    </div>
                                    <ul>
                                        <li>
                                            The user is not aware of all the analysis choices that would affect the conclusions of co-localization analysis
                                        </li>
                                        <li>
                                            The user is not familiar of all the existing methods and tools that support their analysis choices and assumptions
                                        </li>
                                        <li>
                                            The user is uncertain about the association between a pair of tracks, to assess the robustness of findings
                                        </li>
                                        <li>
                                             The user wants to run existing pairwise co-localization analysis tools in one-against-many mode
                                        </li>
                                        <li>
                                             The user wants to run several co-localization analysis tools at once
                                        </li>
                                    </ul>
                                </dd>
                                     <p></p>
                                     <li>
                                <dt><b>I do not know how to get started, please help me</b></dt>
                                <dd>
                                To perform co-localization analysis, you need a query track and reference track of interest in
                                    BED format. Once you have the BED files, upload them using the
                                    <a  class = "button" href="javascript:void(0)" onclick="event.preventDefault(); jQuery('#tool-panel-upload-button', window.parent.jQuery(window.parent.document)).trigger('click');">Upload button</a>
                                    on the top left corner of the web page. If you  do not know how to upload files into a galaxy interface,
                                    follow a <a href='https://galaxyproject.org/tutorials/upload/' target='_top'><font color="#0645AD">quick tutorial</font></a>
                                    Once you upload the files, they will appear in your history (on the panel that appears as margin to the right). To know more
                                    about histories in galaxy, <a href='https://galaxyproject.org/tutorials/histories/' target='_top'><font color="#0645AD">see a tutorial here</font></a>.
                                    Then go to the <a class="button" href="../tool_runner/data_source_redirect?tool_id=hb_conglo_proto_tool">coloc-stats tool</a> and choose the relevant
                                    parameter settings to execute. Once you execute the tool, the analysis will be performed by one or several tools in parallel depending upon the chosen
                                    parameter settings. This may take from few minutes to several hours. The results element will be created in your history, which you can also monitor while
                                    the analysis is in progress by clicking on the “eye” icon on the history element. Once all the computations are completed, the history element will turn green.
                                    If the analysis is failed for some reason, it will turn red.

                                </dd>
                                     </li>
                                <p></p>

                                         <p></p>
                                 <h4>## User interface and sharing results</h4>
                                     <p></p>
                                     <li>
                                <dt><b>What is galaxy project ?</b></dt>
                                <dd>
                                If you are not familiar of galaxy project, you can <a href='https://academic.oup.com/nar/article/44/W1/W3/2499339' target='_top'><font color="#0645AD">read a < 10-sentence abstract </font></a>
                                    and visit the <a href='https://usegalaxy.org/' target='_top'><font color="#0645AD">galaxy project homepage</font></a> to know more about it.
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>I am new to a galaxy interface; give me a quick overview on how to navigate.</b></dt>
                                <dd>
                                You can follow a quick and brief tutorial on <a href='https://galaxyproject.org/tutorials/g101/' target='_top'><font color="#0645AD">using a galaxy interface</font></a> (takes ~ 2 min).
                                </dd>
                                     </li>
                                <p></p>
                                     <li>
                                <dt><b>Is coloc-stats available as a command line tool?</b></dt>
                                <dd>
                                    As of now, coloc-stats is available only through the web server. Command line tool might be a future possibility.
                                </dd>
                                     </li>
                                     <p></p>
                                  <li>
                                <dt><b>Why is it taking ages to run a coloc-stats analysis?</b></dt>
                                <dd>
                                    Depending upon the number of genomic tracks being analysed and also depending upon the number of tools and tool-specific
                                    configuration variants chosen, the analysis might take few minutes to several hours. Especially, some tools by nature are relatively
                                    on a slower side. To quickly confirm that this is not  a server issue, you can run another analysis with less number of genomic tracks
                                    and tools to check if the slowness persists. Depending upon the traffic to the server, sometimes the webpage might appear to be slow and
                                    the computations might be queued. But this is supposed to be a temporary thing and the webpage should return to its original smoothness
                                    within few minutes-hours. If that is not the case <a href='mailto:geirska@ifi.uio.no' target='_top'><font color="#0645AD">you can report to us</font></a> to get support.
                                </dd>
                                     </li>
                                     <p></p>

                                     </li>
                            </dl>
<script type="text/javascript">
    $("#faqs dd").hide();
    $("#faqs dt").click(function () {
        $(this).next("#faqs dd").slideToggle(500);
        $(this).toggleClass("expanded");
    });
</script>

                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <div id="menu3" class="tab-pane fade">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDesc'>
                            <h4>What is coloc-stats</h4>
                        </div>
                        <div class = 'projectDesc'>
                            Coloc-stats is a collaborative project between the authors of several published co-localization analysis tools. The tool is developed at the Department of Informatics at the University of Oslo and the web server is hosted at https://hyperbrowser.uio.no/coloc-stats/
                        </div>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <h4>Tools integrated in coloc-stats</h4>
                        </div>

                        <div class = 'projectDesc'>
                                Coloc-stats provides a unified interface to eight different co-localization analysis tools. Tools are listed in chronological order they are published.

                        </div>
                        <p></p>
                        <div class = 'description'>
                            <a  target="_blank"  href = 'https://doi.org/10.1186/gb-2010-11-12-r121' >Genomic Hyperbrowser</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank" href = 'https://doi.org/10.1093/bioinformatics/bts009' >IntervalStats</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1371/journal.pcbi.1002529' >GenometriCorr</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1016/j.ajhg.2015.05.016' >GoShifter</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1093/bioinformatics/btv612' >LOLA</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = ' https://doi.org/10.1093/gigascience/gix032' >GSuite Hyperbrowser</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1093/bioinformatics/btx379' >Stereogene</a>
                        </div>
                        <div class = 'description'>
                            <a target="_blank"  href = 'https://doi.org/10.1101/157735' >GIGGLE</a>
                        </div>


                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<h4>About us</h4>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'projectDesc'>-->
                            <!--<b>Team at the Department of Informatics, University of Oslo</b>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'description'>-->
                            <!--Geir Kjetil Sandve (Principal Investigator)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Boris Simovski-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Chakravarthi Kanduri-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Sveinung Gundersen-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Dmytro Titov-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Diana Domanska-->
                        <!--</div>-->

                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<b>International collaborators</b>-->
                        <!--</div>-->
                        <!--<p></p>-->
                        <!--<div class = 'description'>-->
                            <!--Christoph Bock (CeMM Research Center for Molecular Medicine of the Austrian Academy of Sciences, Vienna, Austria)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Maria Chikina (University of Pittsburgh School of Medicine, Pittsburgh, PA, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Alexander Favorov (John Hopkins University School of Medicine, Baltimore, Maryland, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Ryan Layer (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Aaron Quinlan (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Nathan Sheffield (Center for Public Health Genomics, University of Virginia, Charlottesville, Virginia, USA)-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--Gosia Trynka (Wellcome Trust Sanger Institute, Hinxton, UK)-->
                        <!--</div>-->

                        <!--<div class = 'projectDesc' style="margin-top:20px">-->
                            <!--<b>Contact us</b>-->
                        <!--</div>-->
                        <!--<div class = 'description'>-->
                            <!--..........-->
                        <!--</div>-->







                    </div>
                </div>
            </div>
        </div>
         <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <h4>About us</h4>
                        </div>
                        <p></p>
                        <div class = 'projectDesc'>
                            <b>Team at the Department of Informatics, University of Oslo</b>
                        </div>
                        <p></p>
                        <div class = 'description'>
                            Geir Kjetil Sandve (Principal Investigator)
                        </div>
                        <div class = 'description'>
                            Boris Simovski
                        </div>
                        <div class = 'description'>
                            Chakravarthi Kanduri
                        </div>
                        <div class = 'description'>
                            Sveinung Gundersen
                        </div>
                        <div class = 'description'>
                            Dmytro Titov
                        </div>
                        <div class = 'description'>
                            Diana Domanska
                        </div>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>International collaborators</b>
                        </div>
                        <p></p>
                        <div class = 'description'>
                            Christoph Bock (CeMM Research Center for Molecular Medicine of the Austrian Academy of Sciences, Vienna, Austria)
                        </div>
                        <div class = 'description'>
                            Maria Chikina (University of Pittsburgh School of Medicine, Pittsburgh, PA, USA)
                        </div>
                        <div class = 'description'>
                            Alexander Favorov (John Hopkins University School of Medicine, Baltimore, Maryland, USA)
                        </div>
                        <div class = 'description'>
                            Ryan Layer (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)
                        </div>
                        <div class = 'description'>
                            Aaron Quinlan (Department of Human Genetics, University of Utah, Salt Lake City, UT, USA)
                        </div>
                        <div class = 'description'>
                            Nathan Sheffield (Center for Public Health Genomics, University of Virginia, Charlottesville, Virginia, USA)
                        </div>
                        <div class = 'description'>
                            Gosia Trynka (Wellcome Trust Sanger Institute, Hinxton, UK)
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>
                <div class="containerNext">
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                <div class="colAll">
                    <div class = 'project'>

                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>Contact us</b>
                        </div>
                        <div class = 'description'>
                            ..........
                        </div>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>How to cite coloc-stats</b>
                        </div>
                        <div class = 'description'>
                            ..........
                        </div>
                        <div class = 'projectDesc' style="margin-top:20px">
                            <b>Version history</b>
                        </div>
                        <div class = 'description'>
                            ..........
                        </div>
                </div>
                </div>
            </div>
        </div>
    </div>



    </div>
</div>
</body>
</html>